--- s/src/WindowManager.cc-	2009-12-16 03:46:07.437022426 +0600
+++ s/src/WindowManager.cc	2009-12-16 03:46:54.855190839 +0600
@@ -840,15 +840,18 @@
 
             case ButtonPress:
                 _screen->setLastEventTime(ev.xbutton.time);
+                ev.xbutton.state &= ~ButtonMotionMask;
                 handleButtonPressEvent(&ev.xbutton);
                 break;
             case ButtonRelease:
                 _screen->setLastEventTime(ev.xbutton.time);
+                ev.xbutton.state &= ~ButtonMotionMask;
                 handleButtonReleaseEvent(&ev.xbutton);
                 break;
 
             case MotionNotify:
                 _screen->setLastEventTime(ev.xmotion.time);
+                ev.xmotion.state &= ~ButtonMotionMask;
                 handleMotionEvent(&ev.xmotion);
                 break;
 
