--- Mesa-6.5.2/include/GL/internal/sarea.h.orig	2007-03-01 05:27:22.000000000 +0800
+++ Mesa-6.5.2/include/GL/internal/sarea.h	2007-03-01 05:27:49.000000000 +0800
@@ -44,6 +44,8 @@
 /* SAREA area needs to be at least a page */
 #if defined(__alpha__)
 #define SAREA_MAX 			0x2000
+#elif defined(__mips__)
+#define SAREA_MAX                       0x4000
 #elif defined(__ia64__)
 #define SAREA_MAX			0x10000		/* 64kB */
 #else
