--- gcc-5250/gcc/collect2.c-	2009-12-06 00:05:33.317131969 +0600
+++ gcc-5250/gcc/collect2.c	2009-12-06 00:05:36.625216967 +0600
@@ -1577,7 +1577,7 @@
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
