--- cups-1.4rc1/Makefile-	2009-08-05 20:19:52.000000000 +0700
+++ cups-1.4rc1/Makefile	2009-08-05 20:20:02.000000000 +0700
@@ -27,7 +27,7 @@
 # Directories to make...
 #
 
-DIRS	=	cups filter backend berkeley cgi-bin driver locale man monitor \
+DIRS	=	cups filter backend berkeley cgi-bin driver locale monitor \
 		notifier ppdc scheduler systemv test \
 		$(PHPDIR) \
 		conf data doc examples $(FONTS) templates
