--- libxslt-1.1.17/libxslt/Makefile.in~	2006-08-18 12:37:06.000000000 +0000
+++ libxslt-1.1.17/libxslt/Makefile.in	2006-08-18 12:37:21.000000000 +0000
@@ -662,7 +662,7 @@
 	@(cd ../xsltproc ; $(MAKE))
 
 install-exec-hook:
-	$(mkinstalldirs) "$(libdir)/libxslt-plugins"
+	$(mkinstalldirs) "$(DESTDIR)$(libdir)/libxslt-plugins"
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
