--- s/configure.orig	2009-10-08 20:18:28.000000000 -0300
+++ s/configure	2010-09-22 12:02:38.923710886 -0300
@@ -21926,10 +21926,6 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-$as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "${sdkdir}/dri.h"; then
   eval "$as_ac_File=yes"
 else
@@ -21954,10 +21950,6 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-$as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "${sdkdir}/sarea.h"; then
   eval "$as_ac_File=yes"
 else
@@ -21982,10 +21974,6 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-$as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "${sdkdir}/dristruct.h"; then
   eval "$as_ac_File=yes"
 else
