--- s/gsf/gsf-utils.c-	2009-02-04 02:31:21.000000000 +0600
+++ s/gsf/gsf-utils.c	2009-02-04 02:31:27.000000000 +0600
@@ -73,7 +73,7 @@
  * mixture.
  */
 #define G_ARMFLOAT_ENDIAN 56781234
-#if defined(__arm__) && !defined(__vfp__) && (G_BYTE_ORDER == G_LITTLE_ENDIAN)
+#if defined(__arm__) && !defined(__VFP_FP__) && (G_BYTE_ORDER == G_LITTLE_ENDIAN)
 #define G_FLOAT_BYTE_ORDER G_ARMFLOAT_ENDIAN
 #else
 #define G_FLOAT_BYTE_ORDER G_BYTE_ORDER
