--- fuse-2.5.3/util/Makefile.in~	2006-10-20 20:43:27.000000000 +0000
+++ fuse-2.5.3/util/Makefile.in	2006-10-20 20:43:49.000000000 +0000
@@ -453,11 +453,6 @@
 install-exec-hook:
 	-chown root $(DESTDIR)$(bindir)/fusermount
 	-chmod u+s $(DESTDIR)$(bindir)/fusermount
-	@if test ! -e $(DESTDIR)/dev/fuse; then \
-		$(mkdir_p) $(DESTDIR)/dev; \
-		echo "mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229"; \
-		mknod $(DESTDIR)/dev/fuse -m 0666 c 10 229; \
-	fi
 
 install-exec-local:
 	$(mkdir_p) $(DESTDIR)$(MOUNT_FUSE_PATH)
