diff -Nur fbpanel-4.12-/config/default.cfg fbpanel-4.12/config/default.cfg
--- fbpanel-4.12-/config/default.cfg	2007-07-31 06:59:09.000000000 +0700
+++ fbpanel-4.12/config/default.cfg	2008-05-16 00:37:22.000000000 +0700
@@ -150,15 +150,15 @@
         }
 
         item {
-            icon = gnome-terminal
-            name = terminal
+            icon = konsole
+            name = Terminal
             #image = PREFIX/share/fbpanel/images/gnome_terminal.svg
-            action = xterm
+            action = aterm
         }
         item {
-            icon = gnome-emacs
-            name = emacs
-            action = emacs
+            icon = leafpad
+            name = TextEdit
+            action = leafpad
         }
 
     }
@@ -185,20 +185,20 @@
     }
 
         button {
-            icon = gnome-terminal
+            icon = konsole
             tooltip = Terminal
-            action = xterm
+            action = aterm
         }
 
     button {
-        icon = mozilla-firefox
-            tooltip = Firefox web browser
-            action =firefox
+        icon = skipstone
+            tooltip = Skipstone web browser
+            action = skipstone
     }
         button {
-            icon = gnome-emacs
-            tooltip = Emacs
-            action = emacs
+            icon = leafpad
+            tooltip = TextEdit
+            action = leafpad
         }
 
     }
