--- s/bin/mklib.orig	2009-06-11 23:46:24.000000000 +0700
+++ s/bin/mklib	2009-08-19 13:09:16.000000000 +0700
@@ -39,7 +39,7 @@
 STATIC=0
 DLOPEN=0
 INSTALLDIR="."
-ARCH="auto"
+ARCH="Linux"
 ARCHOPT=""
 NOPREFIX=0
 EXPORTS=""
@@ -234,9 +234,9 @@
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=${CROSS}g++
             else
-                LINK=gcc
+                LINK=${CROSS}gcc
             fi
 	fi
 
@@ -271,7 +271,7 @@
         elif [ $STATIC = 1 ] ; then
             LIBNAME="lib${LIBNAME}.a"     # prefix with "lib", suffix with ".a"
             echo "mklib: Making" $ARCH "static library: " ${LIBNAME}
-            LINK="ar"
+            LINK=${CROSS}ar
             OPTS="-ru"
             if [ "${ALTOPTS}" ] ; then
                 OPTS=${ALTOPTS}
@@ -285,8 +285,8 @@
 		case $OBJ in
 		    *.a)
 			# extract the .o files from this .a archive
-			FILES=`ar t $OBJ`
-			ar x $OBJ
+			FILES=`${CROSS}ar t $OBJ`
+			${CROSS}ar x $OBJ
 			NEWOBJECTS="$NEWOBJECTS $FILES"
 			# keep track of temporary .o files and delete them below
 			DELETIA="$DELETIA $FILES"
@@ -300,7 +300,7 @@
 
             # make lib
             ${LINK} ${OPTS} ${LIBNAME} ${NEWOBJECTS}
-            ranlib ${LIBNAME}
+            ${CROSS}ranlib ${LIBNAME}
 
 	    # remove temporary extracted .o files
 	    rm -f ${DELETIA}
