--- kexec-tools-1.101/kexec/arch/ppc64/kexec-ppc64.c~	2006-01-24 11:42:55.000000000 +0900
+++ kexec-tools-1.101/kexec/arch/ppc64/kexec-ppc64.c	2006-01-31 11:12:24.000000000 +0900
@@ -37,6 +37,7 @@
 /* Platforms supported by kexec on PPC64 */
 #define PLATFORM_PSERIES	0x0100
 #define PLATFORM_PSERIES_LPAR	0x0101
+#define PLATFORM_PS3		0x0801
 
 static struct exclude_range exclude_range[MAX_MEMORY_RANGES];
 static unsigned long long rmo_top;
@@ -398,6 +399,10 @@
 				if (rmo_top > 0x30000000UL)
 					rmo_top = 0x30000000UL;
 			}
+			if (platform == PLATFORM_PS3) {
+				if (rmo_top > 0x07C00000UL)
+					rmo_top = 0x07C00000UL;
+			}
 			fclose(file);
 			closedir(cdir);
 		} /* memory */
