--- kexec-tools-1.101.org/configure	2005-02-16 21:37:44.000000000 +0900
+++ kexec-tools-1.101/configure	2006-04-10 14:43:49.000000000 +0900
@@ -3825,7 +3836,7 @@
 fi
 
 
-CFLAGS='-Wall -g -fno-strict-aliasing $(CPPFLAGS)'
+CFLAGS='-Wall -O2 -static -fno-strict-aliasing $(CPPFLAGS)'
 BUILD_CFLAGS='-O2 -Wall $(CPPFLAGS)'
 
 if test "$CC"      = "no"; then { { echo "$as_me:$LINENO: error: cc not found" >&5
