diff -puN kexec/arch/i386/kexec-bzImage.c~kexec-x86-fix-bzImage-real-mode-entry kexec/arch/i386/kexec-bzImage.c
--- kexec-tools-1.101-fix-real-mode-entry/kexec/arch/i386/kexec-bzImage.c~kexec-x86-fix-bzImage-real-mode-entry	2005-09-02 11:12:38.000000000 +0530
+++ kexec-tools-1.101-fix-real-mode-entry-vivek/kexec/arch/i386/kexec-bzImage.c	2005-09-02 11:14:18.000000000 +0530
@@ -182,7 +182,7 @@ int do_bzImage_load(struct kexec_info *i
 	/*
 	 * Initialize the 16bit start information.
 	 */
-	regs16.cs = setup_base + 0x20;
+	regs16.cs = (setup_base >> 4) + 0x20;
 	regs16.ip = 0;
 	regs16.ss = (elf_rel_get_addr(&info->rhdr, "stack_end") - 64*1024) >> 4;
 	regs16.esp = 0xFFFC;
diff -puN purgatory/arch/i386/entry32-16.S~kexec-x86-fix-bzImage-real-mode-entry purgatory/arch/i386/entry32-16.S
--- kexec-tools-1.101-fix-real-mode-entry/purgatory/arch/i386/entry32-16.S~kexec-x86-fix-bzImage-real-mode-entry	2005-09-02 11:14:31.000000000 +0530
+++ kexec-tools-1.101-fix-real-mode-entry-vivek/purgatory/arch/i386/entry32-16.S	2005-09-02 11:31:12.000000000 +0530
@@ -25,7 +25,7 @@
 entry16:
 	.code32
 	/* Compute where I am running at */
-	movl	$entry16_debug, %ebx	
+	movl	$entry16, %ebx
 
 	/* Fixup my real mode segment */
 	movl	%ebx, %eax

