--- s/configure.in-	2008-12-29 12:41:32.000000000 +0600
+++ s/configure.in	2008-12-29 12:42:12.000000000 +0600
@@ -139,15 +139,15 @@
 AC_PROG_INSTALL
 
 
-Darwin=`uname`
-if test "$Darwin" = "Darwin" ; then
-   CFLAGS="$CFLAGS -Ddlsym=dlsym_prepend_underscore -no-cpp-precomp"
-   CXXFLAGS="$CXXFLAGS -DNULL=0L"
-   LDFLAGS="$LDFLAGS -flat_namespace -undefined warning"
-   PLUGIN_LDFLAGS="-flat_namespace -bundle -undefined warning"
-else
+#Darwin=`uname`
+#if test "$Darwin" = "Darwin" ; then
+#   CFLAGS="$CFLAGS -Ddlsym=dlsym_prepend_underscore -no-cpp-precomp"
+#   CXXFLAGS="$CXXFLAGS -DNULL=0L"
+#   LDFLAGS="$LDFLAGS -flat_namespace -undefined warning"
+#   PLUGIN_LDFLAGS="-flat_namespace -bundle -undefined warning"
+#else
    PLUGIN_LDFLAGS="-shared"
-fi
+#fi
 
 AC_SUBST(PLUGIN_LDFLAGS)
 
