--- vlc-0.8.5/src/network/rootwrap.c~	2006-07-03 15:02:04.000000000 +0000
+++ vlc-0.8.5/src/network/rootwrap.c	2006-07-03 15:08:48.000000000 +0000
@@ -195,9 +195,11 @@
 
             setsockopt (sock, SOL_SOCKET, SO_REUSEADDR, &val, sizeof (val));
 #ifdef AF_INET6
+#if defined(CONFIG_IPV6_MODULE)
             if (ss.ss_family == AF_INET6)
                 setsockopt (sock, IPPROTO_IPV6, IPV6_V6ONLY, &val, sizeof (val));
 #endif
+#endif
             if (bind (sock, (struct sockaddr *)&ss, len) == 0)
             {
                 send_fd (fd, sock);
