From ae673c2b2c4044e8bbba2c1cb53b1cc0c384afbd Mon Sep 17 00:00:00 2001
From: dann frazier <dannf@hp.com>
Date: Thu, 15 Jan 2009 16:46:48 -0700
Subject: [PATCH] parisc: rtc: Add a missing include for linux/rtc.h

Signed-off-by: dann frazier <dannf@hp.com>
Acked-by: Alessandro Zummo <a.zummo@towertech.it>
Signed-off-by: Kyle McMartin <kyle@mcmartin.ca>
---
 drivers/rtc/rtc-parisc.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/rtc/rtc-parisc.c b/drivers/rtc/rtc-parisc.c
index c6bfa6f..319bb5d 100644
--- a/drivers/rtc/rtc-parisc.c
+++ b/drivers/rtc/rtc-parisc.c
@@ -7,6 +7,7 @@
 #include <linux/module.h>
 #include <linux/time.h>
 #include <linux/platform_device.h>
+#include <linux/rtc.h>
 
 #include <asm/rtc.h>
 
-- 
1.6.0.4

