--- subversion-1.3.1/neon/configure~	2006-08-04 13:06:47.000000000 +0000
+++ subversion-1.3.1/neon/configure	2006-08-04 13:07:00.000000000 +0000
@@ -19151,7 +19151,7 @@
 
 # Library-internal symbols are in the ne__ namespace: tell libtool
 # to not export these from the built library if possible.
-NEON_LINK_FLAGS="$NEON_LINK_FLAGS -export-symbols-regex '^ne_[^_]'"
+NEON_LINK_FLAGS="$NEON_LINK_FLAGS"
 
 # Checks to compile test suite
 echo "$as_me:$LINENO: checking for library containing strerror" >&5
