--- s/configure.in-	2008-12-28 20:50:28.000000000 +0600
+++ s/configure.in	2008-12-28 22:29:35.000000000 +0600
@@ -83,6 +83,9 @@
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [Gettext package.])
 
+LIBDIR=`echo $libdir`
+AC_DEFINE_UNQUOTED(LIBDIR, "/usr/lib", [Libdir])
+
 # take all languages found in file po/LINGUAS
 ALL_LINGUAS="`sed -e '/^#/d' $srcdir/po/LINGUAS`"
 
