--- s/src/Makefile.webkit-	2008-12-29 04:35:51.000000000 +0600
+++ s/src/Makefile.webkit	2008-12-29 04:36:22.000000000 +0600
@@ -10,10 +10,10 @@
 	chmod 755 skipstone
 
 skipstone-bin-webkit: $(OBJS)
-	$(CC) $(OBJS) -o $@ $(LDFLAGS) $(MOZILLA_LIBS)
+	$(CXX) $(OBJS) -o $@ $(LDFLAGS) $(MOZILLA_LIBS)
 
 skipdownload: $(DOWNLOADOBJS)
-	$(CC) $< -o $@ $(GTK_LIBS)
+	$(CXX) $< -o $@ $(GTK_LIBS) $(OPT_LDFLAGS)
 
 clean:
 	(cd ../locale && ${MAKE} clean)
