--- linphone-1.4.1/console/Makefile.in~	2006-09-22 13:08:39.000000000 +0000
+++ linphone-1.4.1/console/Makefile.in	2006-09-22 13:09:15.000000000 +0000
@@ -282,11 +282,11 @@
 
 #in order to workaround a arm-linux cross-compilation libtool bug
 #we need to explicit again the dependencies of the executables:
-@ARMBUILD_TRUE@AM_LDFLAGS = $(top_builddir)/mediastreamer2/src/libquickstream.la \
-@ARMBUILD_TRUE@		$(top_builddir)/mediastreamer2/src/libmediastreamer.la \
-@ARMBUILD_TRUE@		$(top_builddir)/oRTP/src/libortp.la \
-@ARMBUILD_TRUE@		$(SPEEX_LIBS) \
-@ARMBUILD_TRUE@		$(OSIP_LIBS)
+AM_LDFLAGS = $(top_builddir)/mediastreamer2/src/libquickstream.la \
+		$(top_builddir)/mediastreamer2/src/libmediastreamer.la \
+		$(top_builddir)/oRTP/src/libortp.la \
+		$(SPEEX_LIBS) \
+		$(OSIP_LIBS)
 
 all: all-am
 
