--- linux-2.4.32/drivers/tc/zs.c~	2006-06-04 01:02:11.000000000 +0000
+++ linux-2.4.32/drivers/tc/zs.c	2006-06-04 01:03:00.000000000 +0000
@@ -2043,7 +2043,7 @@
 	struct dec_serial *info = &zs_soft[channel];
 
 	if (info->hook) {
-		printk(__FUNCTION__": line %d has already a hook registered\n", channel);
+		printk("%s: line %d has already a hook registered\n", __FUNCTION__,channel);
 
 		return 0;
 	} else {
@@ -2063,8 +2063,8 @@
                 info->hook = NULL;
                 return 1;
         } else {
-                printk(__FUNCTION__": trying to unregister hook on line %d,"
-                       " but none is registered\n", channel);
+                printk("%s: trying to unregister hook on line %d,"
+                       " but none is registered\n", __FUNCTION__, channel);
                 return 0;
         }
 }
