diff -urNp gcc-4.3.2/gcc-orig/config/spu/spu-elf.h gcc-4.3.2/gcc/config/spu/spu-elf.h
--- gcc-4.3.2/gcc-orig/config/spu/spu-elf.h	2008-06-07 14:23:51.000000000 +0200
+++ gcc-4.3.2/gcc/config/spu/spu-elf.h	2008-06-07 14:23:59.000000000 +0200
@@ -25,6 +25,19 @@
 
 
 
+/* Add sysroot prefixes.  The following macros are used only when building
+   the compiler with --with-sysroot.  They have the effect of allowing to
+   build a pair of PPU and SPU cross-compilers with a common sysroot; the
+   SPU compiler will search for its files in ${sysroot}/usr/spu/include
+   and ${sysroot}/usr/spu/lib.  */
+
+#undef STANDARD_STARTFILE_PREFIX_1
+#define STANDARD_STARTFILE_PREFIX_1 ""
+#undef STANDARD_STARTFILE_PREFIX_2
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/spu/lib/"
+#undef STANDARD_INCLUDE_DIR
+#define STANDARD_INCLUDE_DIR "/usr/spu/include"
+
 /* Provide a STARTFILE_SPEC appropriate for GNU/Linux.  Here we add
    the GNU/Linux magical crtbegin.o file (see crtstuff.c) which
    provides part of the support for getting C++ file-scope static
diff -urNp gcc-4.3.2/gcc-orig/config/spu/t-spu-elf gcc-4.3.2/gcc/config/spu/t-spu-elf
--- gcc-4.3.2/gcc-orig/config/spu/t-spu-elf	2008-06-07 14:23:51.000000000 +0200
+++ gcc-4.3.2/gcc/config/spu/t-spu-elf	2008-06-07 14:23:59.000000000 +0200
@@ -14,6 +14,9 @@
 #  along with GCC; see the file COPYING3.  If not see
 #  <http://www.gnu.org/licenses/>.
 
+# Define system directory to match STANDARD_INCLUDE_DIR in spu-elf.h,
+# allowing combined SPU/PPU sysroot builds.
+NATIVE_SYSTEM_HEADER_DIR = /usr/spu/include
 
 # Suppress building libgcc1.a
 LIBGCC1 =
