Subject: use oselas.toolchain toolchains
From: Sascha Hauer <s.hauer@pengutronix.de>

Ensure the right compiler for this project.

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
Signed-off-by: Juergen Beisert <j.beisert@pengutronix.de>

---

Index: Makefile
===================================================================
--- Makefile.orig
+++ Makefile
@@ -125,7 +125,7 @@ ifeq ($(ARCH),ppc)
 CROSS_COMPILE = powerpc-linux-
 endif
 ifeq ($(ARCH),arm)
-CROSS_COMPILE = arm-linux-
+CROSS_COMPILE = arm-iwmmx-linux-gnueabi-
 endif
 ifeq ($(ARCH),i386)
 ifeq ($(HOSTARCH),i386)
