diff -Nur linux-2.6.16/arch/arm/mach-pxa/corgi.c linux-2.6.16-bootargs/arch/arm/mach-pxa/corgi.c
--- linux-2.6.16/arch/arm/mach-pxa/corgi.c	2006-04-30 20:49:18.000000000 +0000
+++ linux-2.6.16-bootargs/arch/arm/mach-pxa/corgi.c	2006-04-30 20:58:30.000000000 +0000
@@ -356,6 +356,7 @@
 MACHINE_START(CORGI, "SHARP Corgi")
 	.phys_io	= 0x40000000,
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
+	.boot_params	= 0xa0000100,
 	.fixup		= fixup_corgi,
 	.map_io		= pxa_map_io,
 	.init_irq	= pxa_init_irq,
@@ -368,6 +369,7 @@
 MACHINE_START(SHEPHERD, "SHARP Shepherd")
 	.phys_io	= 0x40000000,
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
+	.boot_params	= 0xa0000100,
 	.fixup		= fixup_corgi,
 	.map_io		= pxa_map_io,
 	.init_irq	= pxa_init_irq,
@@ -380,6 +382,7 @@
 MACHINE_START(HUSKY, "SHARP Husky")
 	.phys_io	= 0x40000000,
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
+	.boot_params	= 0xa0000100,
 	.fixup		= fixup_corgi,
 	.map_io		= pxa_map_io,
 	.init_irq	= pxa_init_irq,
diff -Nur linux-2.6.16/arch/arm/mach-pxa/poodle.c linux-2.6.16-bootargs/arch/arm/mach-pxa/poodle.c
--- linux-2.6.16/arch/arm/mach-pxa/poodle.c	2006-04-30 20:49:18.000000000 +0000
+++ linux-2.6.16-bootargs/arch/arm/mach-pxa/poodle.c	2006-04-30 20:57:58.000000000 +0000
@@ -423,6 +423,7 @@
 MACHINE_START(POODLE, "SHARP Poodle")
 	.phys_io	= 0x40000000,
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
+	.boot_params	= 0xa0000100,
 	.fixup		= fixup_poodle,
 	.map_io		= pxa_map_io,
 	.init_irq	= pxa_init_irq,
diff -Nur linux-2.6.16/arch/arm/mach-pxa/spitz.c linux-2.6.16-bootargs/arch/arm/mach-pxa/spitz.c
--- linux-2.6.16/arch/arm/mach-pxa/spitz.c	2006-04-30 20:49:18.000000000 +0000
+++ linux-2.6.16-bootargs/arch/arm/mach-pxa/spitz.c	2006-04-30 20:57:42.000000000 +0000
@@ -559,6 +559,7 @@
 MACHINE_START(SPITZ, "SHARP Spitz")
 	.phys_io	= 0x40000000,
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
+	.boot_params	= 0xa0000100,
 	.fixup		= fixup_spitz,
 	.map_io		= pxa_map_io,
 	.init_irq	= pxa_init_irq,
@@ -571,6 +572,7 @@
 MACHINE_START(BORZOI, "SHARP Borzoi")
 	.phys_io	= 0x40000000,
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
+	.boot_params	= 0xa0000100,
 	.fixup		= fixup_spitz,
 	.map_io		= pxa_map_io,
 	.init_irq	= pxa_init_irq,
@@ -583,6 +585,7 @@
 MACHINE_START(AKITA, "SHARP Akita")
 	.phys_io	= 0x40000000,
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
+	.boot_params	= 0xa0000100,
 	.fixup		= fixup_spitz,
 	.map_io		= pxa_map_io,
 	.init_irq	= pxa_init_irq,
diff -Nur linux-2.6.16/arch/arm/mach-pxa/tosa.c linux-2.6.16-bootargs/arch/arm/mach-pxa/tosa.c
--- linux-2.6.16/arch/arm/mach-pxa/tosa.c	2006-04-30 20:49:17.000000000 +0000
+++ linux-2.6.16-bootargs/arch/arm/mach-pxa/tosa.c	2006-04-30 20:58:52.000000000 +0000
@@ -306,6 +306,7 @@
 MACHINE_START(TOSA, "SHARP Tosa")
 	.phys_io	= 0x40000000,
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
+	.boot_params	= 0xa0000100,
 	.fixup          = fixup_tosa,
 	.map_io         = pxa_map_io,
 	.init_irq       = pxa_init_irq,
