--- gcc-4.4.1/gcc/config/spu/spu-elf.h.orig	2009-02-20 21:20:38.000000000 +0600
+++ gcc-4.4.1/gcc/config/spu/spu-elf.h	2009-07-29 16:40:48.000000000 +0700
@@ -42,6 +42,19 @@
 /* We do not provide any "/usr/local/include" directory on SPU.  */
 #undef LOCAL_INCLUDE_DIR
 
+/* Add sysroot prefixes.  The following macros are used only when building
+   the compiler with --with-sysroot.  They have the effect of allowing to
+   build a pair of PPU and SPU cross-compilers with a common sysroot; the
+   SPU compiler will search for its files in ${sysroot}/usr/spu/include
+   and ${sysroot}/usr/spu/lib.  */
+
+#undef STANDARD_STARTFILE_PREFIX_1
+#define STANDARD_STARTFILE_PREFIX_1 ""
+#undef STANDARD_STARTFILE_PREFIX_2
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/spu/lib/"
+#undef STANDARD_INCLUDE_DIR
+#define STANDARD_INCLUDE_DIR "/usr/spu/include"
+
 /* Provide a STARTFILE_SPEC appropriate for GNU/Linux.  Here we add
    the GNU/Linux magical crtbegin.o file (see crtstuff.c) which
    provides part of the support for getting C++ file-scope static
