--- dbus-glib-0.82/tools/Makefile.in-	2009-07-29 03:03:44.000000000 +0700
+++ dbus-glib-0.82/tools/Makefile.in	2009-07-29 03:04:01.000000000 +0700
@@ -522,7 +522,7 @@
 @USE_INTROSPECT_XML_TRUE@dbus-bus-introspect.xml: $(INTROSPECT_XML_PATH)
 @USE_INTROSPECT_XML_TRUE@	cp $(INTROSPECT_XML_PATH) dbus-bus-introspect.xml
 @USE_INTROSPECT_XML_FALSE@dbus-bus-introspect.xml:
-@USE_INTROSPECT_XML_FALSE@	DBUS_TOP_BUILDDIR=$(top_builddir) @DBUS_DAEMONDIR@dbus-daemon --introspect > dbus-bus-introspect.xml.tmp && mv dbus-bus-introspect.xml.tmp dbus-bus-introspect.xml
+@USE_INTROSPECT_XML_FALSE@	DBUS_TOP_BUILDDIR=$(top_builddir) dbus-daemon --introspect > dbus-bus-introspect.xml.tmp && mv dbus-bus-introspect.xml.tmp dbus-bus-introspect.xml
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
