Index: drivers/pcmcia/soc_common.c
===================================================================
--- linux-2.6.12.orig/drivers/pcmcia/soc_common.c	2005-08-16 16:20:42.000000000 +0100
+++ drivers/pcmcia/soc_common.c	2005-08-16 16:21:09.000000000 +0100
@@ -679,7 +679,7 @@
 	/*
 	 * Initialise the per-socket structure.
 	 */
-	for (i = 0; i < nr; i++) {
+	for (i = (nr-1); i >= 0; i--) {
 		skt = &sinfo->skt[i];
 
 		skt->socket.ops = &soc_common_pcmcia_operations;
@@ -767,7 +768,7 @@
 		if (ret)
 			goto out_err_7;
 
-		WARN_ON(skt->socket.sock != i);
+		/* WARN_ON(skt->socket.sock != i); */
 
 		add_timer(&skt->poll_timer);
 
