--- Mesa-7.5/src/egl/drivers/dri/Makefile-	2009-09-09 17:22:18.000000000 +0700
+++ Mesa-7.5/src/egl/drivers/dri/Makefile	2009-09-09 17:22:29.000000000 +0700
@@ -7,7 +7,6 @@
 ### Include directories
 INCLUDE_DIRS = \
 	-I. \
-	-I/usr/include \
 	$(shell pkg-config --cflags-only-I libdrm) \
 	-I$(TOP)/include \
 	-I$(TOP)/include/GL/internal \
--- Mesa-7.5/src/egl/drivers/glx/Makefile-	2009-09-09 17:22:52.000000000 +0700
+++ Mesa-7.5/src/egl/drivers/glx/Makefile	2009-09-09 17:22:56.000000000 +0700
@@ -14,7 +14,6 @@
 
 INCLUDE_DIRS = \
 	-I. \
-	-I/usr/include \
 	$(shell pkg-config --cflags-only-I libdrm) \
 	-I$(TOP)/include \
 	-I$(TOP)/include/GL/internal \
--- Mesa-7.5/src/egl/drivers/xdri/Makefile-	2009-09-09 17:22:43.000000000 +0700
+++ Mesa-7.5/src/egl/drivers/xdri/Makefile	2009-09-09 17:22:47.000000000 +0700
@@ -12,7 +12,6 @@
 
 INCLUDE_DIRS = \
 	-I. \
-	-I/usr/include \
 	$(shell pkg-config --cflags-only-I libdrm) \
 	-I$(TOP)/include \
 	-I$(TOP)/include/GL/internal \
