--- Mesa-7.5/configs/default-	2009-08-20 17:05:41.000000000 +0700
+++ Mesa-7.5/configs/default	2009-08-20 17:05:53.000000000 +0700
@@ -92,7 +92,7 @@
 GALLIUM_DIRS = auxiliary drivers state_trackers
 GALLIUM_AUXILIARY_DIRS = draw translate cso_cache pipebuffer tgsi sct rtasm util indices
 GALLIUM_AUXILIARIES = $(foreach DIR,$(GALLIUM_AUXILIARY_DIRS),$(TOP)/src/gallium/auxiliary/$(DIR)/lib$(DIR).a)
-GALLIUM_DRIVERS_DIRS = softpipe i915simple failover trace
+GALLIUM_DRIVERS_DIRS = softpipe failover trace
 GALLIUM_DRIVERS = $(foreach DIR,$(GALLIUM_DRIVERS_DIRS),$(TOP)/src/gallium/drivers/$(DIR)/lib$(DIR).a)
 GALLIUM_WINSYS_DIRS = xlib egl_xlib
 GALLIUM_WINSYS_DRM_DIRS =
--- Mesa-7.5/src/gallium/winsys/xlib/Makefile-	2009-06-27 01:21:59.000000000 +0700
+++ Mesa-7.5/src/gallium/winsys/xlib/Makefile	2009-08-20 17:09:44.000000000 +0700
@@ -24,8 +24,8 @@
 
 DEFINES += \
 	-DGALLIUM_SOFTPIPE \
-	-DGALLIUM_TRACE \
-	-DGALLIUM_BRW
+	-DGALLIUM_TRACE
+#-DGALLIUM_BRW
 #-DGALLIUM_CELL will be defined by the config */
 
 XLIB_WINSYS_SOURCES = \
