Index: udisks/data/80-udisks.rules
===================================================================
--- udisks.orig/data/80-udisks.rules	2010-03-18 04:16:54.718849374 +0100
+++ udisks/data/80-udisks.rules	2010-03-18 04:17:28.550853564 +0100
@@ -77,11 +77,6 @@
 KERNEL=="dm-*", OPTIONS+="watch"
 
 ##############################################################################################################
-# Probe LVM2 Physical Volumes - this will eventually be part of the LVM2 package
-
-ENV{ID_FS_TYPE}=="LVM2_member", TEST=="/lib/udev/udisks-lvm-pv-export", IMPORT{program}="udisks-lvm-pv-export $env{ID_FS_UUID}"
-
-##############################################################################################################
 
 # Probe for partition tables; this really should be part of udev / util-linux-ng
 #
@@ -104,23 +99,6 @@
 
 ##############################################################################################################
 
-# pick up device-mapper data; this REALLY should be done by rules installed
-# by the device-mapper package
-#
-KERNEL!="dm-*", GOTO="device_mapper_end"
-ACTION!="change", GOTO="device_mapper_end"
-
-ENV{UDISKS_DM_TARGET_TYPES}=="|*error*", GOTO="device_mapper_end"
-
-# avoid probing if it has already been done earlier
-#
-ENV{ID_FS_USAGE}!="", GOTO="device_mapper_end"
-IMPORT{program}="/sbin/blkid -o udev -p $tempnode"
-
-LABEL="device_mapper_end"
-
-##############################################################################################################
-
 # pick up data from MD components; this REALLY should be done by rules installed
 # by mdadm or the kernel package
 #
