--- s/src/device.c-	2010-12-09 18:24:02.709852839 +0600
+++ s/src/device.c	2010-12-09 18:25:07.690609628 +0600
@@ -6307,7 +6307,8 @@
   options = prepend_default_mount_options (fsmo, caller_uid, given_options);
 
   /* validate mount options and check for authorizations */
-  s = g_string_new ("uhelper=udisks,nodev,nosuid");
+  // busybox unlike first option s = g_string_new ("uhelper=udisks,nodev,nosuid");
+  s = g_string_new ("nodev,nosuid");
   for (n = 0; options[n] != NULL; n++)
     {
       const char *option = options[n];
