--- exo-0.5.5/exo-csource/Makefile.in-	2010-12-11 21:33:22.429380275 +0600
+++ exo-0.5.5/exo-csource/Makefile.in	2010-12-11 21:34:33.049440916 +0600
@@ -400,9 +400,9 @@
 	list=`for p in $$list; do echo "$$p"; done | sed 's/$(EXEEXT)$$//'`; \
 	echo " rm -f" $$list; \
 	rm -f $$list
-exo-csource$(EXEEXT): $(exo_csource_OBJECTS) $(exo_csource_DEPENDENCIES) 
+exo-csource$(EXEEXT):
 	@rm -f exo-csource$(EXEEXT)
-	$(AM_V_CCLD)$(exo_csource_LINK) $(exo_csource_OBJECTS) $(exo_csource_LDADD) $(LIBS)
+	gcc main.c -o exo-csource$(EXEEXT) `PKG_CONFIG_PATH=/usr/lib/pkgconfig pkg-config glib-2.0 --cflags --libs`
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
--- exo-0.5.5/exo-csource/main.c-	2010-12-11 21:29:00.349435863 +0600
+++ exo-0.5.5/exo-csource/main.c	2010-12-11 21:32:00.929360101 +0600
@@ -32,18 +32,15 @@
 #include <memory.h>
 #endif
 #include <stdio.h>
-#ifdef HAVE_STDLIB_H
 #include <stdlib.h>
-#endif
-#ifdef HAVE_STRING_H
 #include <string.h>
-#endif
 
 #include <glib.h>
-#include <glib/gi18n-lib.h>
+//#include <glib/gi18n-lib.h>
 #include <glib/gprintf.h>
 
-
+#define _(a) (a)
+#define g_fprintf fprintf
 
 /* --- prototypes --- */
 static void parse_args    (gint        *argc_p,
@@ -300,7 +297,7 @@
 static void
 print_version (void)
 {
-  g_print ("%s %s\n\n", g_get_prgname (), PACKAGE_VERSION);
+  g_print ("%s \n\n", g_get_prgname ());
   g_print (_("Copyright (c) %s\n"
              "        os-cillation e.K. All rights reserved.\n\n"
              "Written by Benedikt Meurer <benny@xfce.org>.\n\n"),
@@ -308,8 +305,7 @@
   g_print (_("%s comes with ABSOLUTELY NO WARRANTY,\n"
              "You may redistribute copies of %s under the terms of\n"
              "the GNU Lesser General Public License which can be found in the\n"
-             "%s source package.\n\n"), g_get_prgname (), g_get_prgname (), PACKAGE_TARNAME);
-  g_print (_("Please report bugs to <%s>.\n"), PACKAGE_BUGREPORT);
+             "source package.\n\n"), g_get_prgname (), g_get_prgname ());
 }
 
 
@@ -325,15 +321,6 @@
   gsize    length;
   gint     n;
 
-  bindtextdomain (GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR);
-
-#ifdef HAVE_BIND_TEXTDOMAIN_CODESET
-  bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
-#endif
-
-  textdomain (GETTEXT_PACKAGE);
-  setlocale (LC_ALL, NULL);
-
   /* set program name */
   g_set_prgname (g_path_get_basename (argv[0]));
 
