--- s/Makefile.in~	2006-08-18 19:19:00.000000000 +0000
+++ s/Makefile.in	2006-08-18 19:19:20.000000000 +0000
@@ -67,7 +67,7 @@
 	uninstall-recursive
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = cut-n-paste po help data pdf ps pixbuf djvu tiff dvi \
+DIST_SUBDIRS = cut-n-paste po data pdf ps pixbuf djvu tiff dvi \
 	comics impress backend lib properties shell thumbnailer
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
@@ -309,7 +309,7 @@
 # Properties (library and application)
 
 # Applications 
-SUBDIRS = cut-n-paste po help data pdf $(am__append_1) $(am__append_2) \
+SUBDIRS = cut-n-paste po data pdf $(am__append_1) $(am__append_2) \
 	$(am__append_3) $(am__append_4) $(am__append_5) \
 	$(am__append_6) $(am__append_7) backend lib properties shell \
 	thumbnailer
