--- linux-2.6.27/init/do_mounts_rd.c-	2009-01-03 23:16:51.000000000 +0600
+++ linux-2.6.27/init/do_mounts_rd.c	2009-01-03 23:30:41.000000000 +0600
@@ -8,6 +8,7 @@
 #include <linux/squashfs_fs.h>
 #include <linux/initrd.h>
 #include <linux/string.h>
+#include <linux/sqmagic.h>
 
 #include "do_mounts.h"
 
@@ -104,7 +105,8 @@
 	}
 
 	/* squashfs is at block zero too */
-	if (squashfsb->s_magic == SQUASHFS_MAGIC) {
+	if ((squashfsb->s_magic == SQUASHFS_MAGIC) ||
+	    (squashfsb->s_magic == SQUASHFS_MAGIC_LZMA)) {
 		printk(KERN_NOTICE
 		       "RAMDISK: squashfs filesystem found at block %d\n",
 		       start_block);
