--- s/glide64/rdp.cpp-	2009-08-17 16:40:50.000000000 +0700
+++ s/glide64/rdp.cpp	2009-08-17 16:42:45.000000000 +0700
@@ -51,6 +51,7 @@
 #ifndef _WIN32
 #include "messagebox.h"
 #include <sys/time.h>
+#include <byteswap.h>
 #endif // _WIN32
 
 char out_buf[2048];
@@ -60,8 +61,8 @@
 BOOL ucode_error_report = TRUE;
 int wrong_tile = -1;
 
-#define BYTESWAP1(s1) asm volatile (" bswap %0; " : "+r" (s1) : :);
-#define BYTESWAP2(s1,s2) asm volatile (" bswap %0; bswap %1; " : "+r" (s1), "+r" (s2) : :);
+#define BYTESWAP1(s1) bswap_32(s1);
+#define BYTESWAP2(s1,s2) bswap_32(s1); bswap_32(s2);
 
 // global strings
 const char *ACmp[4] = { "NONE", "THRESHOLD", "UNKNOWN", "DITHER" };
