--- gtkdialog-0.7.20/src/automaton.c-	2009-11-13 13:29:22.000000000 +0600
+++ gtkdialog-0.7.20/src/automaton.c	2009-11-13 13:31:31.000000000 +0600
@@ -41,6 +41,14 @@
 #define TYPE_PIX 2
 #define TYPE_LABPIX 3
 
+#if defined(__i386__) || defined(__x86_64__)
+    #define GTK_BOX_PACK_START	gtk_box_pack_start
+    #define GTK_BOX_PACK_END	gtk_box_pack_end
+#else
+    #define GTK_BOX_PACK_START	gtk_box_pack_end
+    #define GTK_BOX_PACK_END	gtk_box_pack_start
+#endif
+
 extern gboolean option_no_warning;
 extern gboolean option_centering;
 
@@ -1980,8 +1988,8 @@
 		    	Label = gtk_label_new(attributeset_get_first
 				  (Attr, ATTR_LABEL));
 			gtk_container_add(GTK_CONTAINER(Button), Box);
-			gtk_box_pack_end(GTK_BOX(Box), Label, TRUE, TRUE, 0);
-			gtk_box_pack_end(GTK_BOX(Box), Icon, TRUE, TRUE, 0);
+			GTK_BOX_PACK_END(GTK_BOX(Box), Label, TRUE, TRUE, 0);
+			GTK_BOX_PACK_END(GTK_BOX(Box), Icon, TRUE, TRUE, 0);
 	}
 
 	PIP_DEBUG("");
@@ -2470,11 +2478,11 @@
 				if (s.widgettypes[n] == WIDGET_EDIT ||
 				    s.widgettypes[n] == WIDGET_FRAME ||
 				    s.widgettypes[n] == WIDGET_SCROLLEDW)
-					gtk_box_pack_start(GTK_BOX(Widget),
+					GTK_BOX_PACK_START(GTK_BOX(Widget),
 							   s.widgets[n],
 							   TRUE, TRUE, 0);
 				else
-					gtk_box_pack_start(GTK_BOX(Widget),
+					GTK_BOX_PACK_START(GTK_BOX(Widget),
 							   s.widgets[n],
 							   FALSE, FALSE,
 							   0);
@@ -2500,7 +2508,7 @@
 				    s.widgettypes[n] == WIDGET_FRAME ||
 				    s.widgettypes[n] == WIDGET_SCROLLEDW ||
 				    s.widgettypes[n] == WIDGET_ENTRY)
-					gtk_box_pack_end(GTK_BOX(Widget),
+					GTK_BOX_PACK_END(GTK_BOX(Widget),
 							 s.widgets[n],
 							 TRUE, TRUE, 0);
 				else {
@@ -2508,7 +2516,7 @@
 					 * I can't make it work... I mean to set the expand and fill
 					 * from the dialog description.
 					 */ 
-					gtk_box_pack_end(GTK_BOX(Widget),
+					GTK_BOX_PACK_END(GTK_BOX(Widget),
 							 s.widgets[n],
 							 FALSE, // Expand
 							 FALSE, // Fill
@@ -2536,11 +2544,11 @@
 				if (s.widgettypes[n] == WIDGET_EDIT ||
 				    s.widgettypes[n] == WIDGET_FRAME ||
 				    s.widgettypes[n] == WIDGET_SCROLLEDW)
-					gtk_box_pack_start(GTK_BOX(vbox),
+					GTK_BOX_PACK_START(GTK_BOX(vbox),
 							   s.widgets[n],
 							   TRUE, TRUE, 0);
 				else
-					gtk_box_pack_start(GTK_BOX(vbox),
+					GTK_BOX_PACK_START(GTK_BOX(vbox),
 							   s.widgets[n],
 							   FALSE, FALSE,
 							   0);
