--- openldap-2.3.25/clients/tools/Makefile.in~	2006-08-02 18:19:44.000000000 +0000
+++ openldap-2.3.25/clients/tools/Makefile.in	2006-08-02 18:19:54.000000000 +0000
@@ -98,7 +98,7 @@
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(bindir);				\
 	    done							\
 	)
--- openldap-2.3.25/servers/slapd/Makefile.in~	2006-08-02 18:23:48.000000000 +0000
+++ openldap-2.3.25/servers/slapd/Makefile.in	2006-08-02 18:23:52.000000000 +0000
@@ -375,7 +375,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)/run
-	$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@for i in $(SUBDIRS); do \
 	    if test -d $$i && test -f $$i/Makefile ; then \
--- openldap-2.3.25/servers/slurpd/Makefile.in~	2006-08-02 18:24:28.000000000 +0000
+++ openldap-2.3.25/servers/slurpd/Makefile.in	2006-08-02 18:24:35.000000000 +0000
@@ -60,7 +60,7 @@
 	@-$(INSTALL) -m 700 -d $(DESTDIR)$(localstatedir)/openldap-slurp
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
