--- s/configs/linux-ppc-	2009-08-25 10:20:50.000000000 +0700
+++ s/configs/linux-ppc	2009-08-25 10:21:12.000000000 +0700
@@ -4,6 +4,9 @@
 
 CONFIG_NAME = linux-ppc
 
-OPT_FLAGS = -O3 -mcpu=603 -fsigned-char -funroll-loops
+OPT_FLAGS = -O3 -fsigned-char -funroll-loops
+
+ASM_FLAGS = -DUSE_PPC_ASM -DUSE_VMX_ASM
+MESA_ASM_SOURCES = $(PPC_SOURCES)
 
 # FIXME: Use of PowerPC assembly should be enabled here.
\ В конце файла нет новой строки
