--- Mesa-7.5/src/gallium/winsys/egl_xlib/Makefile-	2009-08-19 16:31:01.000000000 +0700
+++ Mesa-7.5/src/gallium/winsys/egl_xlib/Makefile	2009-08-19 16:31:14.000000000 +0700
@@ -74,7 +74,7 @@
 install: default
 	$(INSTALL) -d $(INSTALL_DIR)/$(LIB_DIR)
 	@if [ -e $(TOP)/$(LIB_DIR) ]; then \
-		$(INSTALL) $(TOP)/$(LIB_DIR)/$(DRIVER_NAME) $(INSTALL_DIR)/$(LIB_DIR); \
+		$(INSTALL) $(TOP)/$(LIB_DIR)/$(DRIVER_NAME) $(DESTDIR)$(INSTALL_DIR)/$(LIB_DIR); \
 	fi
 
 
