--- s/bin/mklib-	2009-01-04 13:51:04.000000000 +0600
+++ s/bin/mklib	2009-01-04 13:51:47.000000000 +0600
@@ -39,7 +39,7 @@
 STATIC=0
 DLOPEN=0
 INSTALLDIR="."
-ARCH="auto"
+ARCH="Linux"
 ARCHOPT=""
 NOPREFIX=0
 EXPORTS=""
@@ -217,9 +217,9 @@
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=${CROSS}g++
             else
-                LINK=gcc
+                LINK=${CROSS}gcc
             fi
 	fi
 
@@ -254,7 +254,7 @@
         elif [ $STATIC = 1 ] ; then
             LIBNAME="lib${LIBNAME}.a"     # prefix with "lib", suffix with ".a"
             echo "mklib: Making" $ARCH "static library: " ${LIBNAME}
-            LINK="ar"
+            LINK=${CROSS}ar
             OPTS="-ru"
             if [ "${ALTOPTS}" ] ; then
                 OPTS=${ALTOPTS}
@@ -262,7 +262,7 @@
             rm -f ${LIBNAME}
             # make lib
             ${LINK} ${OPTS} ${LIBNAME} ${OBJECTS}
-            ranlib ${LIBNAME}
+            ${CROSS}ranlib ${LIBNAME}
             # finish up
             FINAL_LIBS=${LIBNAME}
         else
