--- ekiga-3.2.6/lib/engine/components/ptlib/videoinput-manager-ptlib.cpp-	2009-11-10 02:37:02.000000000 +0600
+++ ekiga-3.2.6/lib/engine/components/ptlib/videoinput-manager-ptlib.cpp	2009-11-10 02:52:45.000000000 +0600
@@ -124,15 +124,16 @@
     error_code = Ekiga::VI_ERROR_DEVICE;
   else if (!input_device->SetVideoFormat (pvideo_format))
     error_code = Ekiga::VI_ERROR_FORMAT;
-  else if (!input_device->SetChannel (current_state.channel))
-    error_code = Ekiga::VI_ERROR_CHANNEL;
-  else if (!input_device->SetColourFormatConverter ("YUV420P"))
-    error_code = Ekiga::VI_ERROR_COLOUR;
-  else if (!input_device->SetFrameRate (current_state.fps))
-    error_code = Ekiga::VI_ERROR_FPS;
-  else if (!input_device->SetFrameSizeConverter (current_state.width, current_state.height, PVideoFrameInfo::eScale))
-    error_code = Ekiga::VI_ERROR_SCALE;
-  else input_device->Start ();
+  else {
+    input_device->SetChannel (current_state.channel);
+    if (!input_device->SetColourFormatConverter ("YUV420P"))
+      error_code = Ekiga::VI_ERROR_COLOUR;
+    else if (!input_device->SetFrameRate (current_state.fps))
+      error_code = Ekiga::VI_ERROR_FPS;
+    else if (!input_device->SetFrameSizeConverter (current_state.width, current_state.height, PVideoFrameInfo::eScale))
+      error_code = Ekiga::VI_ERROR_SCALE;
+    else input_device->Start ();
+  }
 
   if (error_code != Ekiga::VI_ERROR_NONE) {
     PTRACE(1, "GMVideoInputManager_ptlib\tEncountered error " << error_code << " while opening device ");
