--- s/src/osd/sdl/sdl.mak-	2009-08-11 02:18:17.000000000 +0700
+++ s/src/osd/sdl/sdl.mak	2009-08-11 02:18:47.000000000 +0700
@@ -288,10 +288,10 @@
 endif # NO_X11
 
 # make sure we can find X headers
-CFLAGS += -I/usr/X11/include -I/usr/X11R6/include -I/usr/openwin/include
+#CFLAGS += 
 # some systems still put important things in a different prefix
 ifndef NO_X11
-LIBS += -L/usr/X11/lib -L/usr/X11R6/lib -L/usr/openwin/lib
+#LIBS += 
 endif
 endif # Unix
 
