--- leafpad-0.8.16/src/main.c-	2009-01-03 12:58:57.000000000 +0600
+++ leafpad-0.8.16/src/main.c	2009-01-03 12:59:01.000000000 +0600
@@ -305,7 +305,7 @@
 	conf = g_malloc(sizeof(Conf));
 	conf->width       = 600;
 	conf->height      = 400;
-	conf->fontname    = g_strdup("Monospace 12");
+	conf->fontname    = g_strdup("Monospace 10");
 	conf->wordwrap    = FALSE;
 	conf->linenumbers = FALSE;
 	conf->autoindent  = FALSE;
