--- s/configure-	2009-03-24 13:35:30.000000000 +0600
+++ s/configure	2009-03-24 13:35:51.000000000 +0600
@@ -10713,7 +10713,8 @@
 echo "${ECHO_T}no" >&6; }
                 { { echo "$as_me:$LINENO: error: D-Bus library is required" >&5
 echo "$as_me: error: D-Bus library is required" >&2;}
-   { (exit 1); exit 1; }; }
+   #{ (exit 1); exit 1; }; 
+   }
 elif test $pkg_failed = untried; then
 	{ { echo "$as_me:$LINENO: error: D-Bus library is required" >&5
 echo "$as_me: error: D-Bus library is required" >&2;}
--- s/common/error.c-	2009-03-24 13:42:58.000000000 +0600
+++ s/common/error.c	2009-03-24 13:43:12.000000000 +0600
@@ -31,6 +31,8 @@
 #include <string.h>
 #include <errno.h>
 
+#if 0
+
 #include <gdbus.h>
 
 #include "error.h"
@@ -124,3 +126,5 @@
 
 	return DBUS_HANDLER_RESULT_HANDLED;
 }
+
+#endif
--- s/common/error.h-	2009-03-24 13:43:01.000000000 +0600
+++ s/common/error.h	2009-03-24 13:43:22.000000000 +0600
@@ -23,6 +23,8 @@
  *
  */
 
+#if 0
+
 #include <dbus/dbus.h>
 
 #define ERROR_INTERFACE "org.bluez.Error"
@@ -47,3 +49,5 @@
 
 DBusHandlerResult error_common_reply(DBusConnection *conn, DBusMessage *msg,
 					const char *name, const char *descr);
+
+#endif
