--- s/petitboot-paths.h-	2008-01-08 10:12:01.000000000 +0600
+++ s/petitboot-paths.h	2009-01-16 13:52:43.000000000 +0600
@@ -10,13 +10,13 @@
 #endif
 
 #ifndef TMP_DIR
-#define TMP_DIR "/var/tmp/mnt/"
+#define TMP_DIR "/media/"
 #endif
 
-#define PBOOT_DEVICE_SOCKET "/var/tmp/petitboot-dev"
+#define PBOOT_DEVICE_SOCKET "/tmp/petitboot-dev"
 #define MOUNT_BIN "/bin/mount"
 #define UMOUNT_BIN "/bin/umount"
-#define BOOT_GAMEOS_BIN "/usr/bin/ps3-boot-game-os"
+#define BOOT_GAMEOS_BIN "/usr/sbin/ps3-boot-game-os"
 
 /* at present, all default artwork strings are const. */
 #define artwork_pathname(s) (PKG_SHARE_DIR "/artwork/" s)
