--- s/configure-	2009-08-25 15:53:05.000000000 +0700
+++ s/configure	2009-08-25 15:53:19.000000000 +0700
@@ -12174,7 +12174,7 @@
         echo $ECHO_N "(blacklisted) $ECHO_C" >&6
         as_cv_unaligned_access=no
 	;;
-      i?86*|x86_64|amd64|powerpc*|m68k*|cris*)
+      i?86*|x86_64*|amd64*|powerpc*|m68k*|cris*)
         echo $ECHO_N "(whitelisted) $ECHO_C" >&6
         as_cv_unaligned_access=yes
 	;;
