--- s/etc/codecs.conf-	2009-08-22 17:49:24.000000000 +0700
+++ s/etc/codecs.conf	2009-08-22 17:52:04.000000000 +0700
@@ -3620,18 +3620,17 @@
   driver acm
   dll "msadp32.acm"
 
-audiocodec mp3
-  info "mp3lib MPEG layer-2, layer-3"
+audiocodec mad
+  info "libMAD MPEG layer 1-2-3"
   status working
-  comment "Optimized to MMX/SSE/3Dnow!"
-  format 0x50        ; layer-1 && layer-2
-  format 0x55        ; layer-3
+  format 0x50
+  format 0x55
   format 0x5500736d  ; "ms\0\x55" older mp3 fcc (MOV files)
-  format 0x5000736d  ; "ms\0\x50" older mp2 fcc (MOV files)
   fourcc ".mp3"      ; CBR/VBR MP3 (MOV files)
   fourcc "MP3 "      ; used in .nsv files
   fourcc "LAME"      ; used in mythtv .nuv files
-  driver mp3lib
+  driver libmad
+  dll "libmad"
 
 audiocodec ffpcmdaud
   info "D-Cinema audio (FFmpeg)"
@@ -3695,6 +3694,19 @@
   driver ffmpeg
   dll "mp3"
 
+audiocodec mp3
+  info "mp3lib MPEG layer-2, layer-3"
+  status working
+  comment "Optimized to MMX/SSE/3Dnow!"
+  format 0x50        ; layer-1 && layer-2
+  format 0x55        ; layer-3
+  format 0x5500736d  ; "ms\0\x55" older mp3 fcc (MOV files)
+  format 0x5000736d  ; "ms\0\x50" older mp2 fcc (MOV files)
+  fourcc ".mp3"      ; CBR/VBR MP3 (MOV files)
+  fourcc "MP3 "      ; used in .nsv files
+  fourcc "LAME"      ; used in mythtv .nuv files
+  driver mp3lib
+
 audiocodec ffmp3adu
   info "FFmpeg MPEG layer-3 adu audio"
   comment "integer only"
@@ -3713,18 +3725,6 @@
   driver ffmpeg
   dll "mp2"
 
-audiocodec mad
-  info "libMAD MPEG layer 1-2-3"
-  status working
-  format 0x50
-  format 0x55
-  format 0x5500736d  ; "ms\0\x55" older mp3 fcc (MOV files)
-  fourcc ".mp3"      ; CBR/VBR MP3 (MOV files)
-  fourcc "MP3 "      ; used in .nsv files
-  fourcc "LAME"      ; used in mythtv .nuv files
-  driver libmad
-  dll "libmad"
-
 audiocodec mp3acm
   info "MPEG layer-3"
   status working
