--- linux-2.6.17/scripts/Makefile.build~	2006-10-08 17:05:41.000000000 +0000
+++ linux-2.6.17/scripts/Makefile.build	2006-10-08 17:08:24.000000000 +0000
@@ -166,7 +166,7 @@
 cmd_modversions =							\
 	if $(OBJDUMP) -h $(@D)/.tmp_$(@F) | grep -q __ksymtab; then	\
 		$(CPP) -D__GENKSYMS__ $(c_flags) $<			\
-		| $(GENKSYMS) -a $(ARCH)				\
+		| scripts/genksyms/genksyms -a $(ARCH)				\
 		> $(@D)/.tmp_$(@F:.o=.ver);				\
 									\
 		$(LD) $(LDFLAGS) -r -o $@ $(@D)/.tmp_$(@F) 		\
