--- lxpanel-0.3.99/src/plugins/volumealsa/volumealsa.c-	2008-12-30 16:02:41.000000000 +0600
+++ lxpanel-0.3.99/src/plugins/volumealsa/volumealsa.c	2008-12-30 16:25:42.000000000 +0600
@@ -77,12 +77,13 @@
 static void update_display(volume_t* vol)
 {
     /* mute status */
-    snd_mixer_selem_get_playback_switch(vol->master_element, 0, &vol->mute);
+    if (snd_mixer_selem_get_playback_switch(vol->master_element, 0, &vol->mute) < 0)
+	vol->mute = 1;
 
     if (vol->mute==0)
-        gtk_image_set_from_file(vol->tray_icon, ICONS_MUTE);
+    	gtk_image_set_from_file(vol->tray_icon, ICONS_MUTE);
     else
-        gtk_image_set_from_file(vol->tray_icon, ICONS_VOLUME);
+    	gtk_image_set_from_file(vol->tray_icon, ICONS_VOLUME);
 
     g_signal_handler_block( vol->mute_check, vol->mute_handler );
     gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(vol->mute_check), !vol->mute );
@@ -324,7 +325,9 @@
     g_signal_connect( vol->vscale, "scroll-event", G_CALLBACK(on_vscale_scrolled), vol );
 
     vol->mute_check = gtk_check_button_new_with_label(_("Mute"));
-    snd_mixer_selem_get_playback_switch(vol->master_element, 0, &vol->mute);
+    
+    if (snd_mixer_selem_get_playback_switch(vol->master_element, 0, &vol->mute) < 0)
+	gtk_widget_set_sensitive(vol->mute_check, FALSE);
 
     vol->mute_handler = g_signal_connect ((gpointer) vol->mute_check, "toggled",
                                   G_CALLBACK (click_mute),
