--- s/configure.in-	2008-12-29 11:28:25.000000000 +0600
+++ s/configure.in	2008-12-29 11:30:45.000000000 +0600
@@ -78,6 +78,11 @@
 AC_SUBST([SQLITE3_LIBS])
 AC_DEFINE_UNQUOTED(HAVE_SQLITE,$have_sqlite3, [Whether sqlite3 is available])
 
+PKG_CHECK_MODULES(LIBSOUP, libsoup-2.4 >= 2.4, have_libsoup=1, have_libsoup=0)
+AC_SUBST(LIBSOUP_CFLAGS)
+AC_SUBST(LIBSOUP_LIBS)
+AC_DEFINE_UNQUOTED(HAVE_LIBSOUP,$have_libsoup, [Whether libsoup is available])
+
 # i18n
 GETTEXT_PACKAGE=midori
 AC_SUBST(GETTEXT_PACKAGE)
