--- linux-2.6.28/arch/arm/mach-davinci/board-dm644x-evm.c-	2009-03-03 03:03:10.000000000 +0600
+++ linux-2.6.28/arch/arm/mach-davinci/board-dm644x-evm.c	2009-03-03 03:50:17.000000000 +0600
@@ -124,7 +124,7 @@
 static struct davinci_nand_pdata davinci_evm_nandflash_data = {
 	.parts		= davinci_evm_nandflash_partition,
 	.nr_parts	= ARRAY_SIZE(davinci_evm_nandflash_partition),
-	.ecc_mode	= NAND_ECC_HW,
+	.ecc_mode	= NAND_ECC_NONE,
 };
 
 static struct resource davinci_evm_nandflash_resource[] = {
@@ -373,7 +373,7 @@
 	/* irlml6401 switches over 1A, in under 8 msec;
 	 * now it can be managed by nDRV_VBUS ...
 	 */
-	setup_usb(500, 8);
+//	setup_usb(500, 8);
 
 	return 0;
 }
@@ -664,6 +664,8 @@
 			     ARRAY_SIZE(davinci_evm_devices));
 	evm_init_i2c();
 
+	setup_usb(500, 8);
+
 	davinci_setup_mmc(0, &dm6446evm_mmc_config);
 
 	davinci_serial_init(&uart_config);
