diff -Nur linux-2.6.18-orig/drivers/rtc/rtc-sa1100.c linux-2.6.18/drivers/rtc/rtc-sa1100.c
--- linux-2.6.18-orig/drivers/rtc/rtc-sa1100.c	2006-11-05 16:27:41.000000000 +0000
+++ linux-2.6.18/drivers/rtc/rtc-sa1100.c	2006-11-05 17:55:57.000000000 +0000
@@ -189,6 +189,7 @@
 {
 	spin_lock_irq(&sa1100_rtc_lock);
 	RTSR = 0;
+	RTSR = (RTSR_AL|RTSR_HZ);
 	OIER &= ~OIER_E1;
 	OSSR = OSSR_M1;
 	spin_unlock_irq(&sa1100_rtc_lock);
