--- s/src/Makefile.in-	2009-04-17 02:18:39.000000000 +0700
+++ s/src/Makefile.in	2009-04-17 02:22:12.000000000 +0700
@@ -262,7 +262,7 @@
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
 matchbox-keyboard$(EXEEXT): $(matchbox_keyboard_OBJECTS) $(matchbox_keyboard_DEPENDENCIES) 
 	@rm -f matchbox-keyboard$(EXEEXT)
-	$(LINK) $(matchbox_keyboard_LDFLAGS) $(matchbox_keyboard_OBJECTS) $(matchbox_keyboard_LDADD) $(LIBS)
+	$(LINK) $(matchbox_keyboard_LDFLAGS) $(matchbox_keyboard_OBJECTS) $(matchbox_keyboard_LDADD) $(LIBS) -lXrandr
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
