--- xscreensaver-5.00/driver/Makefile.in~	2006-08-18 15:11:09.000000000 +0000
+++ xscreensaver-5.00/driver/Makefile.in	2006-08-18 15:11:26.000000000 +0000
@@ -32,7 +32,7 @@
 CC		= @CC@
 OBJCC		= @OBJCC@
 CFLAGS		= @CFLAGS@
-LDFLAGS		= @LDFLAGS@
+#LDFLAGS		= @LDFLAGS@
 DEFS		= @DEFS@
 INTL_DEFS	= -DLOCALEDIR=\"$(localedir)\"
 SUBP_DEFS	= $(DEFS) -DDEFAULT_PATH_PREFIX='"@HACKDIR@"'
--- xscreensaver-5.00/hacks/Makefile.in~	2006-08-18 15:14:01.000000000 +0000
+++ xscreensaver-5.00/hacks/Makefile.in	2006-08-18 15:14:06.000000000 +0000
@@ -25,7 +25,7 @@
 
 CC		= @CC@
 CFLAGS		= @CFLAGS@
-LDFLAGS		= @LDFLAGS@
+#LDFLAGS		= @LDFLAGS@
 DEFS		= -DSTANDALONE @DEFS@
 LIBS		= @LIBS@
 PERL		= @PERL@
