--- s/src/tint.c-	2009-12-17 17:12:18.466456441 +0600
+++ s/src/tint.c	2009-12-17 17:23:39.516416447 +0600
@@ -141,6 +141,13 @@
 	if (!panel) return;
 
 	if (wm_menu) {
+		pid_t pid;
+		pid = fork();
+		if (pid == 0) {
+			system("/usr/bin/xdotool key super+r");
+			_exit(0);
+		}
+#if 0
 		if ((panel_horizontal && (e->xbutton.x < panel->area.paddingxlr || e->xbutton.x > panel->area.width-panel->area.paddingxlr || e->xbutton.y < panel->area.paddingy || e->xbutton.y > panel->area.paddingy+panel->g_taskbar.height)) || (!panel_horizontal && (e->xbutton.y < panel->area.paddingxlr || e->xbutton.y > panel->area.height-panel->area.paddingxlr || e->xbutton.x < panel->area.paddingy || e->xbutton.x > panel->area.paddingy+panel->g_taskbar.width))) {
 			// forward the click to the desktop window (thanks conky)
 			XUngrabPointer(server.dsp, e->xbutton.time);
@@ -154,6 +161,7 @@
 			XSendEvent(server.dsp, e->xbutton.window, False, ButtonPressMask, e);
 			return;
 		}
+#endif
 	}
 
 	if (e->xbutton.button != 1) return;
