--- s/configure-	2008-12-26 01:08:42.000000000 +0600
+++ s/configure	2008-12-26 01:10:33.000000000 +0600
@@ -4737,7 +4737,7 @@
 # This doesn't mean it didn't use your optimizations.
 
 # this makes HP-UX look like "HP" (sed turns "HP-UX" into "HP" with the -.* pattern)
-OS_NAME=`uname -s | sed "s/\//-/" | sed "s/_/-/" | sed "s/-.*//g"`
+OS_NAME=Linux
 OS_RELEASE=`uname -r | sed "s/\//-/" | sed "s/() .*//g"`
 
 ##################################################################
