--- linux-2.6.27/scripts/genksyms/parse.y-	2008-12-12 02:21:04.000000000 +0600
+++ linux-2.6.27/scripts/genksyms/parse.y	2008-12-12 02:21:38.000000000 +0600
@@ -24,7 +24,11 @@
 %{
 
 #include <assert.h>
+#ifdef __APPLE__
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #include "genksyms.h"
 
 static int is_typedef;
--- linux-2.6.27/scripts/genksyms/parse.c_shipped-	2008-12-12 02:28:08.000000000 +0600
+++ linux-2.6.27/scripts/genksyms/parse.c_shipped	2008-12-12 02:28:57.000000000 +0600
@@ -160,7 +160,11 @@
 
 
 #include <assert.h>
+#ifdef __APPLE__
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #include "genksyms.h"
 
 static int is_typedef;
@@ -318,7 +322,11 @@
 #   elif defined _AIX
 #    define YYSTACK_ALLOC __alloca
 #   elif defined _MSC_VER
-#    include <malloc.h> /* INFRINGES ON USER NAME SPACE */
+#    ifdef __APPLE__
+#     include <stdlib.h>
+#    else
+#     include <malloc.h>
+#    endif
 #    define alloca _alloca
 #   else
 #    define YYSTACK_ALLOC alloca
