--- ./netwerk/socket/base/nsUDPSocketProvider.h.orig	2009-01-08 16:23:30.000000000 +0800
+++ ./netwerk/socket/base/nsUDPSocketProvider.h	2009-01-08 16:23:35.000000000 +0800
@@ -45,7 +45,6 @@
     NS_DECL_ISUPPORTS
     NS_DECL_NSISOCKETPROVIDER
 
-private:
     ~nsUDPSocketProvider();
 
 };
--- ./layout/base/nsLayoutHistoryState.cpp.orig	2009-01-08 19:41:07.000000000 +0800
+++ ./layout/base/nsLayoutHistoryState.cpp	2009-01-08 19:41:15.000000000 +0800
@@ -59,10 +59,10 @@
   NS_IMETHOD GetState(const nsCString& aKey, nsPresState** aState);
   NS_IMETHOD RemoveState(const nsCString& aKey);
   NS_IMETHOD_(PRBool) HasStates() const;
+  ~nsLayoutHistoryState() {}
 
 
 private:
-  ~nsLayoutHistoryState() {}
 
   nsClassHashtable<nsCStringHashKey,nsPresState> mStates;
 };
--- ./widget/src/gtk2/nsImageToPixbuf.h.orig	2009-01-08 18:09:05.000000000 +0800
+++ ./widget/src/gtk2/nsImageToPixbuf.h	2009-01-08 18:13:20.000000000 +0800
@@ -56,10 +56,10 @@
                                           PRInt32 aWidth, PRInt32 aHeight);
         static GdkPixbuf* PatternToPixbuf(gfxPattern* aPattern,
                                           PRInt32 aWidth, PRInt32 aHeight);
+        ~nsImageToPixbuf() {}
     private:
         static GdkPixbuf* ImgSurfaceToPixbuf(gfxImageSurface* aImgSurface,
                                              PRInt32 aWidth, PRInt32 aHeight);
-        ~nsImageToPixbuf() {}
 };
 
 
--- ./widget/src/gtk2/nsNativeKeyBindings.h.orig	2009-01-08 18:13:33.000000000 +0800
+++ ./widget/src/gtk2/nsNativeKeyBindings.h	2009-01-08 18:13:59.000000000 +0800
@@ -90,8 +90,8 @@
                                    DoCommandCallback aCallback,
                                    void *aCallbackData);
 
-private:
   ~nsNativeKeyBindings() NS_HIDDEN;
+private:
 
   PRBool KeyPressInternal(const nsNativeKeyEvent& aEvent,
                           DoCommandCallback aCallback,
