--- s/configure-	2010-12-11 21:05:57.219359910 +0600
+++ s/configure	2010-12-11 21:06:20.189369860 +0600
@@ -13583,42 +13583,9 @@
 rm -f conftest.mmap conftest.txt
 
 
-if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-  #include <string.h>
-  #include <time.h>
-  int
-  main (int argc, char **argv)
-  {
-    struct tm tm;
-    char      buffer[16];
-    tm.tm_year = 81;
-    if (strftime (buffer, 16, "%EY", &tm) == 4 && strcmp (buffer, "1981") == 0)
-      return 0;
-    return 1;
-  }
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-
-
 $as_echo "#define HAVE_STRFTIME_EXTENSION 1" >>confdefs.h
 
 
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
-
-
     GETTEXT_PACKAGE=exo-1
 
 cat >>confdefs.h <<_ACEOF
