diff -Nur ptxdist-1.99.2/generic/etc/init.d/networking ptxdist-1.99.2-new/generic/etc/init.d/networking
--- ptxdist-1.99.2/generic/etc/init.d/networking	2008-02-07 22:00:36.000000000 +0600
+++ ptxdist-1.99.2-new/generic/etc/init.d/networking	2008-05-10 14:14:18.000000000 +0700
@@ -5,6 +5,7 @@
 cat /proc/cmdline | grep nfsroot > /dev/null
 
 if [ $? != 0 ]; then
+	ifdown -a
 	ifup -a
 else
 	# We are doing nfsroot. We cannot simply call ifup -a
diff -Nur ptxdist-1.99.2/rules/other/Toplevel.make ptxdist-1.99.2-new/rules/other/Toplevel.make
--- ptxdist-1.99.2/rules/other/Toplevel.make	2008-02-07 21:59:30.000000000 +0600
+++ ptxdist-1.99.2-new/rules/other/Toplevel.make	2008-05-05 12:59:26.000000000 +0700
@@ -306,7 +306,7 @@
 #
 # Define what images should be build
 #
-SEL_ROOTFS-y				:=
+#SEL_ROOTFS-y				:=
 SEL_ROOTFS-$(PTXCONF_IMAGE_TGZ)		+= $(IMAGEDIR)/root.tgz
 SEL_ROOTFS-$(PTXCONF_IMAGE_JFFS2)	+= $(IMAGEDIR)/root.jffs2
 SEL_ROOTFS-$(PTXCONF_IMAGE_EXT2)	+= $(IMAGEDIR)/root.ext2
