--- tslib-07072006/tests/ts_calibrate.c~	2010-01-16 20:42:22.000000000 -0500
+++ tslib-07072006/tests/ts_calibrate.c	2010-01-16 20:47:49.000000000 -0500
@@ -182,7 +182,7 @@
 #ifdef USE_INPUT_API
 		ts = ts_open("/dev/input/event0", 0);
 #else
-		ts = ts_open("/dev/touchscreen/ucb1x00", 0);
+		ts = ts_open("/dev/input/event1", 0);
 #endif /* USE_INPUT_API */
 	}
 
--- tslib-07072006/tests/ts_harvest.c~	2010-01-16 20:42:32.000000000 -0500
+++ tslib-07072006/tests/ts_harvest.c	2010-01-16 20:48:37.000000000 -0500
@@ -78,7 +78,7 @@
 #ifdef USE_INPUT_API
 		tsdevice = strdup ("/dev/input/event0");
 #else
-		tsdevice = strdup ("/dev/touchscreen/ucb1x00");
+		tsdevice = strdup ("/dev/input/event1");
 #endif /* USE_INPUT_API */
         }
 
--- tslib-07072006/tests/ts_print.c~	2010-01-16 20:42:42.000000000 -0500
+++ tslib-07072006/tests/ts_print.c	2010-01-16 20:49:05.000000000 -0500
@@ -31,7 +31,7 @@
 #ifdef USE_INPUT_API
                 ts = ts_open("/dev/input/event0", 0);
 #else
-                ts = ts_open("/dev/touchscreen/ucb1x00", 0);
+                ts = ts_open("/dev/input/event1", 0);
 #endif /* USE_INPUT_API */
         }
 
--- tslib-07072006/tests/ts_print_raw.c~	2010-01-16 20:43:23.000000000 -0500
+++ tslib-07072006/tests/ts_print_raw.c	2010-01-16 20:49:26.000000000 -0500
@@ -31,7 +31,7 @@
 #ifdef USE_INPUT_API
                 ts = ts_open("/dev/input/event0", 0);
 #else
-                ts = ts_open("/dev/touchscreen/ucb1x00", 0);
+                ts = ts_open("/dev/input/event1", 0);
 #endif /* USE_INPUT_API */
         }
 
--- tslib-07072006/tests/ts_test.c~	2010-01-16 20:43:40.000000000 -0500
+++ tslib-07072006/tests/ts_test.c	2010-01-16 20:49:46.000000000 -0500
@@ -123,7 +123,7 @@
 #ifdef USE_INPUT_API
 		tsdevice = strdup ("/dev/input/event0");
 #else
-		tsdevice = strdup ("/dev/touchscreen/ucb1x00");
+		tsdevice = strdup ("/dev/input/event1");
 #endif /* USE_INPUT_API */
         }
 
