--- mplayerplug-in-3.55/Source/plugin-support.cpp-	2010-01-18 17:01:35.685885368 +0600
+++ mplayerplug-in-3.55/Source/plugin-support.cpp	2010-01-18 17:03:21.136615235 +0600
@@ -121,7 +121,7 @@
     if (len == 0)
 	return NULL;
     filename = (char *) NPN_MemAlloc(sizeof(char) * (len + 1));
-    tmp = rindex(url, '/');
+    tmp = rindex((char *) url, '/');
     if (tmp == NULL) {
 	strcpy(filename, url);
 	return filename;
