diff -Nur squashfs3.4/squashfs-tools/mksquashfs.c squashfs3.4-applefix/squashfs-tools/mksquashfs.c
--- squashfs3.4/squashfs-tools/mksquashfs.c	2008-08-26 14:01:39.000000000 +0700
+++ squashfs3.4-applefix/squashfs-tools/mksquashfs.c	2008-12-19 15:58:03.000000000 +0600
@@ -63,6 +63,10 @@
 #include "global.h"
 #include "sort.h"
 
+#ifndef FNM_EXTMATCH
+#define FNM_EXTMATCH 0
+#endif
+
 #ifdef SQUASHFS_TRACE
 #define TRACE(s, args...)	do { \
 					if(progress_enabled) \
diff -Nur squashfs3.4/squashfs-tools/unsquashfs.c squashfs3.4-applefix/squashfs-tools/unsquashfs.c
--- squashfs3.4/squashfs-tools/unsquashfs.c	2008-08-26 14:01:54.000000000 +0700
+++ squashfs3.4-applefix/squashfs-tools/unsquashfs.c	2008-12-19 16:02:01.000000000 +0600
@@ -52,6 +52,7 @@
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
+#include <sys/sysctl.h>
 #else
 #include <endian.h>
 #endif
@@ -60,6 +61,10 @@
 #include "read_fs.h"
 #include "global.h"
 
+#ifndef FNM_EXTMATCH
+#define FNM_EXTMATCH 0
+#endif
+
 #ifdef SQUASHFS_TRACE
 #define TRACE(s, args...)		do { \
 						pthread_mutex_lock(&screen_mutex); \
