--- s/src/Unix/sigsegv.cpp-	2010-08-13 23:45:40.239751285 +0300
+++ s/src/Unix/sigsegv.cpp	2010-08-13 23:45:56.999760583 +0300
@@ -1610,7 +1610,11 @@
 
 // This function handles the badaccess to memory.
 // It is called from the signal handler or the exception handler.
+#ifdef HAVE_SIGSEGV_RECOVERY
 static bool handle_badaccess(SIGSEGV_FAULT_HANDLER_ARGLIST_1)
+#else
+static bool handle_badaccess()
+#endif
 {
 #ifdef HAVE_MACH_EXCEPTIONS
 	// We must match the initial count when writing back the CPU state registers
