--- s/Makefile.in-	2008-12-19 05:13:31.000000000 +0600
+++ s/Makefile.in	2008-12-19 05:14:20.000000000 +0600
@@ -208,7 +208,7 @@
 themedir = @themedir@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = 8x8 16x16 22x22 24x24 32x32 48x48 scalable po
+SUBDIRS = 8x8 16x16 22x22 24x24 32x32 48x48 scalable
 pkgconfigdir = $(datadir)/pkgconfig
 pkgconfig_DATA = gnome-icon-theme.pc
 THEME_IN_FILES = index.theme.in
@@ -633,7 +633,7 @@
 
 info-am:
 
-install-data-am: install-data-local install-pkgconfigDATA \
+install-data-am: \
 	install-themeDATA
 
 install-dvi: install-dvi-recursive
