diff -Nur linux-2.6.18-orig/drivers/input/mousedev.c linux-2.6.18/drivers/input/mousedev.c
--- linux-2.6.18-orig/drivers/input/mousedev.c	2006-11-05 16:27:39.000000000 +0000
+++ linux-2.6.18/drivers/input/mousedev.c	2006-11-05 18:10:08.000000000 +0000
@@ -736,12 +736,14 @@
 		.evbit = { BIT(EV_KEY) | BIT(EV_REL) },
 		.relbit = { BIT(REL_WHEEL) },
 	},	/* A separate scrollwheel */
+#if 0
 	{
 		.flags = INPUT_DEVICE_ID_MATCH_EVBIT | INPUT_DEVICE_ID_MATCH_KEYBIT | INPUT_DEVICE_ID_MATCH_ABSBIT,
 		.evbit = { BIT(EV_KEY) | BIT(EV_ABS) },
 		.keybit = { [LONG(BTN_TOUCH)] = BIT(BTN_TOUCH) },
 		.absbit = { BIT(ABS_X) | BIT(ABS_Y) },
 	},	/* A tablet like device, at least touch detection, two absolute axes */
+#endif
 	{
 		.flags = INPUT_DEVICE_ID_MATCH_EVBIT | INPUT_DEVICE_ID_MATCH_KEYBIT | INPUT_DEVICE_ID_MATCH_ABSBIT,
 		.evbit = { BIT(EV_KEY) | BIT(EV_ABS) },
@@ -754,6 +756,15 @@
 
 MODULE_DEVICE_TABLE(input, mousedev_ids);
 
+static struct input_device_id mousedev_blacklist[] = {
+	{
+		.flags = INPUT_DEVICE_ID_MATCH_EVBIT | INPUT_DEVICE_ID_MATCH_KEYBIT | INPUT_DEVICE_ID_MATCH_ABSBIT,
+		.evbit = { BIT(EV_KEY) | BIT(EV_ABS) },
+		.keybit = { [LONG(BTN_TOUCH)] = BIT(BTN_TOUCH) },
+	}, 	/* Avoid itouchpads, touchscreens and tablets */
+	{ }, 	/* Terminating entry */
+};
+
 static struct input_handler mousedev_handler = {
 	.event =	mousedev_event,
 	.connect =	mousedev_connect,
@@ -762,6 +773,7 @@
 	.minor =	MOUSEDEV_MINOR_BASE,
 	.name =		"mousedev",
 	.id_table =	mousedev_ids,
+	.blacklist = 	mousedev_blacklist,
 };
 
 #ifdef CONFIG_INPUT_MOUSEDEV_PSAUX
