--- linux-2.6.18/fs/jffs2/scan.c~	2006-09-20 03:42:06.000000000 +0000
+++ linux-2.6.18/fs/jffs2/scan.c	2006-11-07 21:36:40.000000000 +0000
@@ -640,8 +640,8 @@
 			inbuf_ofs = ofs - buf_ofs;
 			while (inbuf_ofs < buf_len) {
 				if (*(uint32_t *)(&buf[inbuf_ofs]) != 0xffffffff) {
-					printk(KERN_WARNING "Empty flash at 0x%08x ends at 0x%08x\n",
-					       empty_start, ofs);
+//					printk(KERN_WARNING "Empty flash at 0x%08x ends at 0x%08x\n",
+//					       empty_start, ofs);
 					if ((err = jffs2_scan_dirty_space(c, jeb, ofs-empty_start)))
 						return err;
 					goto scan_more;
