--- kexec-tools-2.0.1/kexec/kexec-elf-exec.c.orig	2008-10-31 08:00:38.000000000 +0600
+++ kexec-tools-2.0.1/kexec/kexec-elf-exec.c	2010-06-16 12:17:02.251843998 +0700
@@ -112,6 +112,9 @@
 
 	}
 
+	/* Update entry point to reflect new load address*/
+	ehdr->e_entry += base;
+
 	/* Read in the PT_LOAD segments */
 	for(i = 0; i < ehdr->e_phnum; i++) {
 		struct mem_phdr *phdr;
@@ -127,10 +130,9 @@
 		add_segment(info,
 			phdr->p_data, size,
 			phdr->p_paddr + base, phdr->p_memsz);
+		base = phdr->p_paddr + base;
 	}
 
-	/* Update entry point to reflect new load address*/
-	ehdr->e_entry += base;
 
 	result = 0;
  out:
