--- /dev/null	2009-08-07 12:44:29.000000000 +0700
+++ s/mtpaint.desktop	2009-08-12 15:35:01.000000000 +0700
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Version=1.0
+Name=mtPaint graphic editor
+GenericName=Painting program
+Comment=painting program to create pixel art and manipulate digital photos
+Comment[es_ES]=programa de pintura para crear pixel art y manipular fotos digitales
+Comment[pt_BR]=programa de pintura para criar pixel art e manipular fotos digitais
+Type=Application
+Exec=mtpaint
+Icon=mtpaint
+Categories=Graphics;RasterGraphics;
