--- s/Makefile.in~	2006-10-09 04:43:14.000000000 +0000
+++ s/Makefile.in	2006-10-09 04:47:50.000000000 +0000
@@ -115,20 +115,20 @@
 	$(SRCDIR)/script/user-group $(USERNAME) $(GROUPNAME) $(ANSWERALL) $(SRCDIR)
 
 # check if the directories we use exist, and if not, create them
-	if test ! -d $(DESTSBIN); then $(INSTALL) -g $(ROOTGROUP) -o $(ROOTNAME) -m 755 -d $(DESTSBIN) ; fi
-	if test ! -d $(DESTBIN); then $(INSTALL) -g $(ROOTGROUP) -o $(ROOTNAME) -m 755 -d $(DESTBIN) ; fi
-	if test ! -d $(ETC); then $(INSTALL) -g $(ROOTGROUP) -o $(ROOTNAME) -m 755 -d $(ETC) ; fi
+	if test ! -d $(DESTSBIN); then $(INSTALL) -m 755 -d $(DESTSBIN) ; fi
+	if test ! -d $(DESTBIN); then $(INSTALL) -m 755 -d $(DESTBIN) ; fi
+	if test ! -d $(ETC); then $(INSTALL) -m 755 -d $(ETC) ; fi
 # create the spool dir (and change its mode if it already exists)
-	$(INSTALL) -g $(GROUPNAME) -o $(USERNAME) -m 770 -d $(FCRONTABS)
+	$(INSTALL) -m 770 -d $(FCRONTABS)
 
-	$(INSTALL) -g $(ROOTGROUP) -o $(ROOTNAME) -m 110 -s fcron $(DESTSBIN)
-	$(INSTALL) -g $(GROUPNAME) -o $(USERNAME) -m $(BINMODE) -s fcrontab $(DESTBIN)
-	$(INSTALL) -g $(GROUPNAME) -o $(ROOTNAME) -m $(BINMODESIGHUP) -s fcronsighup $(DESTBIN)
+	$(INSTALL) -m 110 fcron $(DESTSBIN)
+	$(INSTALL) -m $(BINMODE) fcrontab $(DESTBIN)
+	$(INSTALL) -m $(BINMODESIGHUP) fcronsighup $(DESTBIN)
 ifeq ($(FCRONDYN), 1)
-	$(INSTALL) -g $(GROUPNAME) -o $(USERNAME) -m $(BINMODE) -s fcrondyn $(DESTBIN)
+	$(INSTALL) -m $(BINMODE) fcrondyn $(DESTBIN)
 endif
-	test -f $(ETC)/fcron.allow || test -f $(ETC)/fcron.deny || $(INSTALL) -m 640 -o $(ROOTNAME) -g $(GROUPNAME) $(SRCDIR)/files/fcron.allow $(SRCDIR)/files/fcron.deny $(ETC)
-	test -f $(ETC)/fcron.conf || $(INSTALL) -m 640 -o $(ROOTNAME) -g $(GROUPNAME) files/fcron.conf $(ETC)
+	test -f $(ETC)/fcron.allow || test -f $(ETC)/fcron.deny || $(INSTALL) -m 640 $(SRCDIR)/files/fcron.allow $(SRCDIR)/files/fcron.deny $(ETC)
+	test -f $(ETC)/fcron.conf || $(INSTALL) -m 640 files/fcron.conf $(ETC)
 ifeq ($(USEPAM), 1)
 	$(SRCDIR)/script/install-pam-conf $(SRCDIR) $(ETC) $(ROOTNAME) $(ROOTGROUP) "$(INSTALL)"
 endif
@@ -138,18 +138,18 @@
 # in order to get correct rights when upgrading :
 # Note : we don't use "chown user:group file" because some systems use ":"
 #        and others "." as separator.
-	find $(FCRONTABS) -type f \( -name "*.orig" -a ! -name "$(ROOTNAME).orig" -a ! -name "$(SYSFCRONTAB).orig" \) -exec chown $(USERNAME) {} \; -exec chgrp $(GROUPNAME) {} \; -exec chmod 640 {} \;
-	find $(FCRONTABS) -type f \( -name "$(ROOTNAME).orig" -o -name "$(SYSFCRONTAB).orig" \) -exec chown $(ROOTNAME) {} \; -exec chgrp $(GROUPNAME) {} \; -exec chmod 600 {} \;
-	find $(FCRONTABS) -type f ! -name "*.orig" -exec chown $(ROOTNAME) {} \; -exec chgrp $(ROOTGROUP) {} \; -exec chmod 600 {} \;
-	if test -f $(ETC)/fcron.deny; then chown $(ROOTNAME) $(ETC)/fcron.deny ; chgrp $(GROUPNAME) $(ETC)/fcron.deny ; fi
-	if test -f $(ETC)/fcron.allow; then chown $(ROOTNAME) $(ETC)/fcron.allow ; chgrp $(GROUPNAME) $(ETC)/fcron.allow ; fi
+	#find $(FCRONTABS) -type f \( -name "*.orig" -a ! -name "$(ROOTNAME).orig" -a ! -name "$(SYSFCRONTAB).orig" \) -exec chown $(USERNAME) {} \; -exec chgrp $(GROUPNAME) {} \; -exec chmod 640 {} \;
+	#find $(FCRONTABS) -type f \( -name "$(ROOTNAME).orig" -o -name "$(SYSFCRONTAB).orig" \) -exec chown $(ROOTNAME) {} \; -exec chgrp $(GROUPNAME) {} \; -exec chmod 600 {} \;
+	#find $(FCRONTABS) -type f ! -name "*.orig" -exec chown $(ROOTNAME) {} \; -exec chgrp $(ROOTGROUP) {} \; -exec chmod 600 {} \;
+	#if test -f $(ETC)/fcron.deny; then chown $(ROOTNAME) $(ETC)/fcron.deny ; chgrp $(GROUPNAME) $(ETC)/fcron.deny ; fi
+	#if test -f $(ETC)/fcron.allow; then chown $(ROOTNAME) $(ETC)/fcron.allow ; chgrp $(GROUPNAME) $(ETC)/fcron.allow ; fi
 
 ifeq ($(BOOTINSTALL), 1)
-	$(SRCDIR)/script/boot-install "$(INSTALL) -o $(ROOTNAME)" $(DESTSBIN) $(DEBUG) $(FCRONTABS) $(ANSWERALL) $(SRCDIR)
+	$(SRCDIR)/script/boot-install "$(INSTALL)"  $(DESTSBIN) $(DEBUG) $(FCRONTABS) $(ANSWERALL) $(SRCDIR)
 endif
 
 install-boot: install
-	$(SRCDIR)/script/boot-install "$(INSTALL) -o $(ROOTNAME)" $(DESTSBIN) $(DEBUG) $(FCRONTABS)  $(ANSWERALL) $(SRCDIR)
+	$(SRCDIR)/script/boot-install "$(INSTALL)" $(DESTSBIN) $(DEBUG) $(FCRONTABS)  $(ANSWERALL) $(SRCDIR)
 
 install-restart: install
 ifeq ($(DEBUG), 1)
--- s/doc/Makefile.in~	2006-10-09 04:51:39.000000000 +0000
+++ s/doc/Makefile.in	2006-10-09 04:52:36.000000000 +0000
@@ -104,25 +104,25 @@
 		fi ; \
 		for i in 1 3 5 8; do \
 			if test ! -d $$DIR/man$$i; then \
-	$(INSTALL) -g $(ROOTGROUP) -o $(ROOTNAME) -m 755 -d $$DIR/man$$i ; \
+	$(INSTALL) -m 755 -d $$DIR/man$$i ; \
 			 fi ; \
-	$(INSTALL) -m 644 -o $(ROOTNAME) $(SRCDIR)/$$l/man/*.$$i $$DIR/man$$i ; \
+	$(INSTALL) -m 644 $(SRCDIR)/$$l/man/*.$$i $$DIR/man$$i ; \
 		done ; \
 	done )
 
 	@(echo "Installing documentation files in $(DESTDOC)/fcron-$(VERSION)...")
-	@(if test ! -d $(DESTDOC)/fcron-$(VERSION); then $(INSTALL) -g $(ROOTGROUP) -o $(ROOTNAME) -m 755 -d $(DESTDOC)/fcron-$(VERSION) ; fi)
+	@(if test ! -d $(DESTDOC)/fcron-$(VERSION); then $(INSTALL) -m 755 -d $(DESTDOC)/fcron-$(VERSION) ; fi)
 
 	@(for l in $(LANGUAGES); do \
 		DIR=$(DESTDOC)/fcron-$(VERSION) ; \
 		if test ! -d $$DIR/$$l; then \
-	$(INSTALL) -g $(ROOTGROUP) -o $(ROOTNAME) -m 755 -d $$DIR/$$l ; \
+	$(INSTALL) -m 755 -d $$DIR/$$l ; \
 		fi ; \
 		for i in txt HTML; do \
 			if test ! -d $$DIR/$$l/$$i; then \
-	$(INSTALL) -g $(ROOTGROUP) -o $(ROOTNAME) -m 755 -d $$DIR/$$l/$$i ; \
+	$(INSTALL) -m 755 -d $$DIR/$$l/$$i ; \
 			 fi ; \
-	$(INSTALL) -m 644 -o $(ROOTNAME) $$l/$$i/* $$DIR/$$l/$$i/ ;\
+	$(INSTALL) -m 644 $$l/$$i/* $$DIR/$$l/$$i/ ;\
 		done ; \
 	done )
 
