--- ../xx/xserver-xorg-video-siliconmotion-1.7.3/src/smi_driver.c	2010-03-09 23:39:23.451798382 +0800
+++ xserver-xorg-video-siliconmotion-1.7.3/src/smi_driver.c	2010-03-10 10:36:58.835798871 +0800
@@ -1805,6 +1797,11 @@
     pScreen->SaveScreen = SMI_SaveScreen;
     pSmi->CloseScreen = pScreen->CloseScreen;
     pScreen->CloseScreen = SMI_CloseScreen;
+        /* Added by Belcon to enable LCD Panel Control Select */
+        if (pSmi->Chipset == SMI_LYNXEMplus)
+        {
+                VGAOUT8_INDEX (pSmi, VGA_SEQ_INDEX, VGA_SEQ_DATA, 0x22, 2);
+        }
 
     if ((IS_MSOC(pSmi) &&
 	 !xf86DPMSInit(pScreen, SMI501_DisplayPowerManagementSet, 0)) ||
