--- qt-x11-opensource-src-4.4.3/mkspecs/common-cross/linux.conf-	2009-01-22 13:07:24.000000000 +0600
+++ qt-x11-opensource-src-4.4.3/mkspecs/common-cross/linux.conf	2009-01-22 13:11:44.000000000 +0600
@@ -7,12 +7,12 @@
 
 QMAKE_INCDIR          =
 QMAKE_LIBDIR          =
-QMAKE_INCDIR_X11      = /usr/X11R6/include
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib
+QMAKE_INCDIR_X11      = @TARGET_DIR@/include
+QMAKE_LIBDIR_X11      = @TARGET_DIR@/lib
 QMAKE_INCDIR_QT       = $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT       = $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL   = /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL   = @TARGET_DIR@/include
+QMAKE_LIBDIR_OPENGL   = @TARGET_DIR@/lib
 
 QMAKE_LIBS            =
 QMAKE_LIBS_DYNLOAD    = -ldl
@@ -26,9 +26,9 @@
 QMAKE_MOC             = $$[QT_INSTALL_BINS]/moc
 QMAKE_UIC             = $$[QT_INSTALL_BINS]/uic
 
-QMAKE_AR              = ar cqs
-QMAKE_OBJCOPY         = objcopy
-QMAKE_RANLIB          =
+QMAKE_AR              = @CROSS@ar cqs
+QMAKE_OBJCOPY         = @CROSS@objcopy
+QMAKE_RANLIB          = @CROSS@ranlib
 
 QMAKE_TAR             = tar -cf
 QMAKE_GZIP            = gzip -9f
