--- s/configure-	2009-08-22 16:44:42.000000000 +0700
+++ s/configure	2009-08-22 16:46:52.000000000 +0700
@@ -2029,22 +2029,6 @@
     fi
 fi
 
-
-die_license_disabled() {
-    enabled $1 || { enabled $2 && die "$2 is $1 and --enable-$1 is not specified."; }
-}
-
-die_license_disabled gpl libfaad2
-die_license_disabled gpl libx264
-die_license_disabled gpl libxvid
-die_license_disabled gpl postproc
-die_license_disabled gpl x11grab
-
-die_license_disabled nonfree libfaac
-
-die_license_disabled version3 libopencore_amrnb
-die_license_disabled version3 libopencore_amrwb
-
 enabled version3 && { enabled gpl && enable gplv3 || enable lgplv3; }
 
 check_deps $ARCH_EXT_LIST
@@ -2128,6 +2112,8 @@
 EOF
     fi
 
+    add_cflags -fPIC
+
 elif enabled sparc; then
 
     enabled vis && check_asm vis '"pdist %f0, %f0, %f0"' -mcpu=ultrasparc &&
