--- s/configure~	2003-12-05 13:46:53.000000000 +0000
+++ s/configure	2006-09-29 11:16:26.000000000 +0000
@@ -2699,7 +2699,7 @@
 fi
 
 
-if test "$cross_compiling" = yes; then
+if false; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -2748,7 +2748,7 @@
 rm -f core core.* *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -2795,7 +2795,7 @@
 rm -f core core.* *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -4158,7 +4158,7 @@
 
 { echo "$as_me:$LINENO: checking fifos..." >&5
 echo "$as_me: checking fifos..." >&6;}
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -4267,7 +4267,7 @@
 if test -n "$fifo"; then
 { echo "$as_me:$LINENO: checking for broken fifo implementation..." >&5
 echo "$as_me: checking for broken fifo implementation..." >&6;}
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -4360,7 +4360,7 @@
 
 { echo "$as_me:$LINENO: checking sockets..." >&5
 echo "$as_me: checking sockets..." >&6;}
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -4455,7 +4455,7 @@
 if test -n "$sock"; then
 { echo "$as_me:$LINENO: checking socket implementation..." >&5
 echo "$as_me: checking socket implementation..." >&6;}
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -4568,7 +4568,7 @@
 
 { echo "$as_me:$LINENO: checking select return value..." >&5
 echo "$as_me: checking select return value..." >&6;}
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -4910,7 +4910,7 @@
 fi
 rm -f conftest.$ac_objext conftest$ac_exeext conftest.$ac_ext
 
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -5354,7 +5354,7 @@
 { echo "$as_me:$LINENO: checking default tty permissions/group..." >&5
 echo "$as_me: checking default tty permissions/group..." >&6;}
 rm -f conftest_grp
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -6201,7 +6201,7 @@
 echo "$as_me: checking for avenrun symbol..." >&6;}
   nlist64=
   for av in avenrun _avenrun _Loadavg avenrun _avenrun _Loadavg; do
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -6477,7 +6477,7 @@
 rm -f conftest.$ac_objext conftest$ac_exeext conftest.$ac_ext
 { echo "$as_me:$LINENO: checking signal implementation..." >&5
 echo "$as_me: checking signal implementation..." >&6;}
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -6999,7 +6999,7 @@
 
 { echo "$as_me:$LINENO: checking whether memcpy/memmove/bcopy handles overlapping arguments..." >&5
 echo "$as_me: checking whether memcpy/memmove/bcopy handles overlapping arguments..." >&6;}
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -7051,7 +7051,7 @@
 rm -f core core.* *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -7105,7 +7105,7 @@
 fi
 
 
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -7946,7 +7946,7 @@
 fi
 test -n "$seqptx" && LIBS="-ltermcap -lc -lsocket -linet -lnsl -lsec -lseq"
 
-if test "$cross_compiling" = yes; then
+if test "$cross_compiling" = xxx; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
