diff -NPur linux-2.6.27.4.orig/include/linux/squashfs_fs.h linux-2.6.27.4/include/linux/squashfs_fs.h
--- linux-2.6.27.4.orig/include/linux/squashfs_fs.h	2008-10-29 18:03:31.000000000 -0500
+++ linux-2.6.27.4/include/linux/squashfs_fs.h	2008-10-29 18:04:01.000000000 -0500
@@ -341,7 +341,7 @@
 	unsigned int		offset:13;
 	unsigned int		type:3;
 	unsigned int		size:8;
-	int			inode_number:16;
+	signed int		inode_number:16; /* very important signedness */
 	char			name[0];
 } __attribute__ ((packed));
 
