--- gtk+-2.10.1/gtk/Makefile.in~	2006-08-16 17:51:37.000000000 +0000
+++ gtk+-2.10.1/gtk/Makefile.in	2006-08-16 17:51:59.000000000 +0000
@@ -2927,11 +2927,11 @@
 	done									\
 	&& touch stamp-icons
 
-gtkbuiltincache.h: @REBUILD@ stamp-icons
-	$(MAKE) $(AM_MAKEFLAGS) gtk-update-icon-cache
-	GDK_PIXBUF_MODULE_FILE=$(top_builddir)/gdk-pixbuf/gdk-pixbuf.loaders	\
-	   ./gtk-update-icon-cache --force --ignore-theme-index 		\
-	   --source builtin_icons stock-icons > gtkbuiltincache.h
+#gtkbuiltincache.h: @REBUILD@ stamp-icons
+#	$(MAKE) $(AM_MAKEFLAGS) gtk-update-icon-cache
+#	GDK_PIXBUF_MODULE_FILE=$(top_builddir)/gdk-pixbuf/gdk-pixbuf.loaders	\
+#	   ./gtk-update-icon-cache --force --ignore-theme-index 		\
+#	   --source builtin_icons stock-icons > gtkbuiltincache.h
 
 install-data-local:
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
