--- s/Makefile.in-	2009-08-20 21:40:33.000000000 +0700
+++ s/Makefile.in	2009-08-20 21:40:38.000000000 +0700
@@ -66,7 +66,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(pkgconfigdir)"
 pkgconfigDATA_INSTALL = $(INSTALL_DATA)
-DATA = $(noinst_DATA) $(pkgconfig_DATA)
+DATA = $(pkgconfig_DATA)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 ETAGS = etags
@@ -198,7 +198,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src examples doc
+SUBDIRS = src doc
 ACLOCAL_AMFLAGS = -I m4
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libmtp.pc
