--- s/Makefile.in-	2009-01-09 20:25:52.000000000 +0600
+++ s/Makefile.in	2009-08-31 20:52:01.000000000 +0700
@@ -590,18 +590,7 @@
   tmac \
   src/utils/afmtodit \
   src/roff/grog \
-  src/roff/nroff \
-  doc \
-  contrib/mm \
-  contrib/chem \
-  contrib/pic2graph \
-  contrib/eqn2graph \
-  contrib/grap2graph \
-  contrib/groffer \
-  contrib/mom \
-  contrib/hdtbl \
-  contrib/pdfmark \
-  contrib/gdiffmk
+  src/roff/nroff
 ALLDIRS=$(INCDIRS) $(LIBDIRS) $(PROGDIRS) \
   $(DEVDIRS) $(XDEVDIRS) $(OTHERDEVDIRS) $(TTYDEVDIRS) $(OTHERDIRS)
 EXTRADIRS=\
