--- fontconfig-2.6.0/fc-arch/Makefile.am-	2008-01-02 22:44:39.000000000 +0600
+++ fontconfig-2.6.0/fc-arch/Makefile.am	2008-12-11 01:21:20.000000000 +0600
@@ -24,7 +24,7 @@
 
 CC = @CC_FOR_BUILD@
 EXEEXT = @EXEEXT_FOR_BUILD@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) -o $@
 
 INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(WARN_CFLAGS)
 
--- fontconfig-2.6.0/fc-arch/Makefile.in-	2008-06-01 09:30:58.000000000 +0700
+++ fontconfig-2.6.0/fc-arch/Makefile.in	2008-12-11 01:21:30.000000000 +0600
@@ -242,7 +242,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) -o $@
 INCLUDES = -I${top_srcdir}/src -I${top_srcdir} $(WARN_CFLAGS)
 TMPL = fcarch.tmpl.h
 STMPL = ${top_srcdir}/fc-arch/fcarch.tmpl.h
