--- linphone-1.3.5/gnome/Makefile.am~	2006-09-22 11:19:00.000000000 +0000
+++ linphone-1.3.5/gnome/Makefile.am	2006-09-22 11:19:16.000000000 +0000
@@ -75,6 +75,10 @@
 
 endif			
 
+AM_LDFLAGS = $(top_builddir)/oRTP/src/libortp.la \
+				$(SPEEX_LIBS) \
+				$(OSIP_LIBS)
+
 EXTRA_DIST = 						\
 	GNOME_LinphoneApplet.server.in.in	\
 	applet.c \
--- linphone-1.3.5/gnome/Makefile.in~	2006-09-22 11:19:03.000000000 +0000
+++ linphone-1.3.5/gnome/Makefile.in	2006-09-22 11:20:56.000000000 +0000
@@ -111,6 +111,9 @@
 @AMDEP_TRUE@	./$(DEPDIR)/linphone_applet-support.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/main.Po ./$(DEPDIR)/presence.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/propertybox.Po ./$(DEPDIR)/support.Po
+AM_LDFLAGS = $(top_builddir)/oRTP/src/libortp.la \
+				$(SPEEX_LIBS) \
+				$(OSIP_LIBS)
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 LTCOMPILE = $(LIBTOOL) --mode=compile $(CC) $(DEFS) \
