--- odcctools-758-20091206/include/foreign/mach/machine/vm_types.h-	2009-12-07 00:17:44.837747550 +0600
+++ odcctools-758-20091206/include/foreign/mach/machine/vm_types.h	2009-12-07 00:19:35.486467695 +0600
@@ -6,8 +6,14 @@
 typedef uint32_t        natural_t;
 typedef int32_t         integer_t;
 
+#if defined(__LP64__)
+typedef uint64_t	uintptr_t;
+typedef uintptr_t       vm_offset_t;
+typedef uintptr_t       vm_size_t;
+#else
 typedef natural_t       vm_offset_t;
 typedef natural_t       vm_size_t;
+#endif
 
 typedef uint64_t        mach_vm_address_t;
 typedef uint64_t        mach_vm_offset_t;
@@ -16,6 +22,4 @@
 typedef vm_offset_t             pointer_t;
 typedef vm_offset_t             vm_address_t;
 
-
-
 #endif
