--- s/configure-	2009-03-16 08:35:05.000000000 +0100
+++ s/configure	2009-03-16 08:38:10.000000000 +0100
@@ -24810,16 +24810,6 @@
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
   fi
 
-  if test "$OPENSSL_ENABLED" = "1"; then
-    if test -n "$LIB_OPENSSL"; then
-
-       LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$LIB_OPENSSL"
-       export LD_LIBRARY_PATH
-       { echo "$as_me:$LINENO: Added $LIB_OPENSSL to LD_LIBRARY_PATH" >&5
-echo "$as_me: Added $LIB_OPENSSL to LD_LIBRARY_PATH" >&6;}
-    fi
-  fi
-
 fi
 
 
@@ -25432,15 +25422,6 @@
    { (exit 1); exit 1; }; }
   fi
 
-  if test "$LIBSSH2_ENABLED" = "1"; then
-    if test -n "$LIB_LIBSSH2"; then
-
-       LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$LIB_LIBSSH2"
-       export LD_LIBRARY_PATH
-       { echo "$as_me:$LINENO: Added $LIB_LIBSSH2 to LD_LIBRARY_PATH" >&5
-echo "$as_me: Added $LIB_LIBSSH2 to LD_LIBRARY_PATH" >&6;}
-    fi
-  fi
 fi
 
 
@@ -25708,13 +25689,6 @@
         { echo "$as_me:$LINENO: detected GnuTLS version $version" >&5
 echo "$as_me: detected GnuTLS version $version" >&6;}
 
-        if test -n "$gtlslib"; then
-
-          LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$gtlslib"
-          export LD_LIBRARY_PATH
-          { echo "$as_me:$LINENO: Added $gtlslib to LD_LIBRARY_PATH" >&5
-echo "$as_me: Added $gtlslib to LD_LIBRARY_PATH" >&6;}
-        fi
       fi
 
     fi
@@ -25978,10 +25952,6 @@
 echo "$as_me: detected NSS version $version" >&6;}
 
 
-        LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$nssprefix/lib$libsuff"
-        export LD_LIBRARY_PATH
-        { echo "$as_me:$LINENO: Added $nssprefix/lib$libsuff to LD_LIBRARY_PATH" >&5
-echo "$as_me: Added $nssprefix/lib$libsuff to LD_LIBRARY_PATH" >&6;}
       fi
 
     fi
