--- s/Makefile-	2009-08-12 04:29:38.000000000 +0700
+++ s/Makefile	2009-08-12 04:32:34.000000000 +0700
@@ -16,6 +16,9 @@
 
 usplash_BACKEND_LDFLAGS = -ldl
 
+BDFTOBOGL:=./bogl/bdftobogl
+PNGTOBOGL:=./bogl/pngtobogl
+
 ifeq ($(BACKEND), svga)
 	BACKEND_dir = bogl svgalib
 	usplash_BACKEND = usplash_svga.o usplash_bogl.o
@@ -59,10 +62,10 @@
 	$(COMPILE) -o $@ -c $<
 
 .png.c:
-	./bogl/pngtobogl $< > $@
+	$(PNGTOBOGL) $< > $@
 
 .bdf.c:
-	./bogl/bdftobogl $< > $@
+	$(BDFTOBOGL) $< > $@
 
 
 install:
@@ -72,9 +75,11 @@
 	$(INSTALL_PROGRAM) usplash_down $(DESTDIR)/sbin
 	$(INSTALL_PROGRAM) update-usplash-theme $(DESTDIR)/usr/sbin
 	$(INSTALL_PROGRAM) libusplash.so.0 $(DESTDIR)/lib/libusplash.so.0
-	ln -sf /lib/libusplash.so.0 $(DESTDIR)/lib/libusplash.so
+	ln -sf libusplash.so.0 $(DESTDIR)/lib/libusplash.so
+ifneq (1,$(CROSS))
 	$(INSTALL_PROGRAM) bogl/pngtobogl $(DESTDIR)/usr/bin/pngtousplash
 	$(INSTALL_PROGRAM) bogl/bdftobogl $(DESTDIR)/usr/bin/bdftousplash
+endif
 	$(INSTALL_DATA) usplash-theme.h $(DESTDIR)/usr/include/usplash-theme.h
 	$(INSTALL_DATA) usplash_backend.h $(DESTDIR)/usr/include/usplash_backend.h
 	$(INSTALL_DATA) libusplash.h $(DESTDIR)/usr/include/libusplash.h
