--- s/configure-	2005-07-11 19:43:27.000000000 +0700
+++ s/configure	2009-12-18 19:23:10.275206029 +0600
@@ -23720,7 +23720,8 @@
 Value of bitfield test can't find out for cross-compiling and we
 don't know its value for host \"$host\".
 *********************************************************************" >&2;}
-   { (exit 1); exit 1; }; }
+   #{ (exit 1); exit 1; }; 
+   }
   esac
 
 else
@@ -23760,7 +23761,8 @@
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+#   { (exit 1); exit 1; };
+   }
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -23794,7 +23796,8 @@
 ( exit $ac_status )
 { { echo "$as_me:$LINENO: error: unsupported bitfield ordering" >&5
 echo "$as_me: error: unsupported bitfield ordering" >&2;}
-   { (exit 1); exit 1; }; }
+   #{ (exit 1); exit 1; }; 
+   }
 fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
