--- s/configure.in-	2010-08-15 09:44:39.000000000 +0200
+++ s/configure.in	2010-08-15 09:44:47.000000000 +0200
@@ -73,15 +73,6 @@
 	AC_MSG_RESULT([yes])
 fi
 
-if test ${enable_doc} = yes; then
-	GNOME_DOC_INIT
-else
-	dnl Do not care if GDU is not found
-	GNOME_DOC_INIT(,,[:])
-	dnl Set   #define HAVE_DISABLE_DOC 1   to indicate documentation not wanted
-	AC_DEFINE([HAVE_DISABLE_DOC], [1], [Define to 1 if --disable-doc specified])
-fi
-
 AM_CONDITIONAL(DISABLE_DOC, test ${enable_doc} = no)
 
 
