diff -Nur data-orig/ethdown.sh data/ethdown.sh
--- data-orig/ethdown.sh	2009-01-02 14:22:25.000000000 +0600
+++ data/ethdown.sh	2009-01-02 14:22:46.000000000 +0600
@@ -1,3 +1,3 @@
 #!/bin/sh
-dhcpcd --release $LXNM_IFNAME
-ifconfig $LXNM_IFNAME down
+
+ifdown $LXNM_IFNAME
diff -Nur data-orig/ethrepair.sh data/ethrepair.sh
--- data-orig/ethrepair.sh	2009-01-02 14:22:25.000000000 +0600
+++ data/ethrepair.sh	2009-01-02 14:23:12.000000000 +0600
@@ -1,4 +1,4 @@
 #!/bin/sh
-ifconfig $LXNM_IFNAME down
-ifconfig $LXNM_IFNAME up
-dhcpcd --renew $LXNM_IFNAME
+
+ifdown $LXNM_IFNAME
+ifup $LXNM_IFNAME
diff -Nur data-orig/ethup.sh data/ethup.sh
--- data-orig/ethup.sh	2009-01-02 14:22:25.000000000 +0600
+++ data/ethup.sh	2009-01-02 14:24:29.000000000 +0600
@@ -1,3 +1,3 @@
 #!/bin/sh
-ifconfig $LXNM_IFNAME up
-dhcpcd --renew $LXNM_IFNAME
+
+ifup $LXNM_IFNAME
diff -Nur data-orig/wificonnect.sh data/wificonnect.sh
--- data-orig/wificonnect.sh	2009-01-02 14:22:25.000000000 +0600
+++ data/wificonnect.sh	2009-01-02 14:30:53.000000000 +0600
@@ -1,19 +1,19 @@
 #!/bin/sh
 # <ifname> <essid> <en_type> <password> <bssid>
 
+ifdown $LXNM_IFNAME
+
 if [ A"$LXNM_WIFI_PROTO" = A"NONE" ]; then
 	# without encryption
-	ifconfig $LXNM_IFNAME up
 	if [ A"$LXNM_WIFI_ESSID" = A ]; then
 		iwconfig $LXNM_IFNAME essid "$LXNM_WIFI_ESSID" key off
 	else
 		iwconfig $LXNM_IFNAME ap "$LXNM_WIFI_APADDR" key off
 	fi
 
-	dhcpcd --renew $LXNM_IFNAME
+	ifup $LXNM_IFNAME
 elif [ A"$LXNM_WIFI_PROTO" = A"WEP" ]; then
 	# WEP
-	ifconfig $LXNM_IFNAME up
 	if [ ! A"$LXNM_WIFI_ESSID" = A ]; then
 		if [ `echo "$LXNM_WIFI_KEY" | wc -c` = 6 ] || [ `echo "$LXNM_WIFI_KEY" | wc -c` = 14 ]; then
 			iwconfig $LXNM_IFNAME essid "$LXNM_WIFI_ESSID" key "s:$LXNM_WIFI_KEY"
@@ -22,7 +22,7 @@
 		fi
 	fi
 
-	dhcpcd --renew $LXNM_IFNAME
+	ifup $LXNM_IFNAME
 else
 	# start trying to associate with the WPA network using SSID test.
 	wpa_supplicant -g/var/run/wpa_supplicant-global -B
@@ -39,5 +39,5 @@
 	wpa_cli -i$LXNM_IFNAME set_network 0 proto "$LXNM_WIFI_PROTO"
 	wpa_cli -i$LXNM_IFNAME enable_network 0
 
-	dhcpcd --renew $LXNM_IFNAME
+	ifup $LXNM_IFNAME
 fi
diff -Nur data-orig/wifidown.sh data/wifidown.sh
--- data-orig/wifidown.sh	2009-01-02 14:22:25.000000000 +0600
+++ data/wifidown.sh	2009-01-02 14:28:37.000000000 +0600
@@ -1,3 +1,3 @@
 #!/bin/sh
-dhcpcd --release $LXNM_IFNAME
-ifconfig $LXNM_IFNAME down
+
+ifdown $LXNM_IFNAME
diff -Nur data-orig/wifirepair.sh data/wifirepair.sh
--- data-orig/wifirepair.sh	2009-01-02 14:22:25.000000000 +0600
+++ data/wifirepair.sh	2009-01-02 14:29:00.000000000 +0600
@@ -1,3 +1,4 @@
 #!/bin/sh
-ifconfig $LXNM_IFNAME down
-ifconfig $LXNM_IFNAME up
+
+ifdown $LXNM_IFNAME
+ifup $LXNM_IFNAME
diff -Nur data-orig/wifiup.sh data/wifiup.sh
--- data-orig/wifiup.sh	2009-01-02 14:22:25.000000000 +0600
+++ data/wifiup.sh	2009-01-02 14:29:11.000000000 +0600
@@ -1,3 +1,3 @@
 #!/bin/sh
-ifconfig $LXNM_IFNAME up
-dhcpcd --renew $LXNM_IFNAME
+
+ifup $LXNM_IFNAME
