--- dbus-glib-0.71/dbus/examples/statemachine/Makefile.am~	2006-08-24 17:07:48.000000000 +0000
+++ dbus-glib-0.71/dbus/examples/statemachine/Makefile.am	2006-08-24 17:13:34.000000000 +0000
@@ -19,10 +19,10 @@
 BUILT_SOURCES = statemachine-server-glue.h statemachine-glue.h
 
 statemachine-server-glue.h: statemachine-server.xml
-	$(LIBTOOL) --mode=execute $(top_builddir)/dbus/dbus-binding-tool --prefix=sm_server --mode=glib-server --output=$@ $<
+	$(LIBTOOL) --mode=execute $(HOST_DBUS-BINDING-TOOL)/dbus-binding-tool --prefix=sm_server --mode=glib-server --output=$@ $<
 
 statemachine-glue.h: statemachine.xml
-	$(LIBTOOL) --mode=execute $(top_builddir)/dbus/dbus-binding-tool --prefix=sm_object --mode=glib-server --output=$@ $<
+	$(LIBTOOL) --mode=execute $(HOST_DBUS-BINDING-TOOL)/dbus-binding-tool --prefix=sm_object --mode=glib-server --output=$@ $<
 
 sm-marshal.c: Makefile sm-marshal.list
 	@GLIB_GENMARSHAL@ --prefix=sm_marshal $(srcdir)/sm-marshal.list --header --body > $@.tmp && mv $@.tmp $@
--- dbus-glib-0.71/dbus/examples/statemachine/Makefile.in~	2006-08-24 17:07:51.000000000 +0000
+++ dbus-glib-0.71/dbus/examples/statemachine/Makefile.in	2006-08-24 17:14:00.000000000 +0000
@@ -523,10 +523,10 @@
 
 
 statemachine-server-glue.h: statemachine-server.xml
-	$(LIBTOOL) --mode=execute $(top_builddir)/dbus/dbus-binding-tool --prefix=sm_server --mode=glib-server --output=$@ $<
+	$(LIBTOOL) --mode=execute $(HOST_DBUS-BINDING-TOOL)/dbus-binding-tool --prefix=sm_server --mode=glib-server --output=$@ $<
 
 statemachine-glue.h: statemachine.xml
-	$(LIBTOOL) --mode=execute $(top_builddir)/dbus/dbus-binding-tool --prefix=sm_object --mode=glib-server --output=$@ $<
+	$(LIBTOOL) --mode=execute $(HOST_DBUS-BINDING-TOOL)/dbus-binding-tool --prefix=sm_object --mode=glib-server --output=$@ $<
 
 sm-marshal.c: Makefile sm-marshal.list
 	@GLIB_GENMARSHAL@ --prefix=sm_marshal $(srcdir)/sm-marshal.list --header --body > $@.tmp && mv $@.tmp $@
