--- s/src/lib/edje_private.h-	2010-12-10 03:50:34.539822599 +0600
+++ s/src/lib/edje_private.h	2010-12-10 03:53:19.311255095 +0600
@@ -31,9 +31,16 @@
 #include <errno.h>
 #include <libgen.h>
 
-#include <lua.h>
-#include <lualib.h>
-#include <lauxlib.h>
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+#include "lua.h"
+#include "lualib.h"
+#include "lauxlib.h"
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
 #include <setjmp.h>
 
 #ifndef _MSC_VER
