--- licq-1.3.4-RC1/plugins/qt-gui/src/optionsdlg.h~	2006-09-21 20:17:50.000000000 +0000
+++ licq-1.3.4-RC1/plugins/qt-gui/src/optionsdlg.h	2006-09-21 20:18:02.000000000 +0000
@@ -24,7 +24,7 @@
 # define DEFAULT_URL_VIEWER tr("KDE default")
 class KURLRequester;
 #else
-# define DEFAULT_URL_VIEWER "viewurl-mozilla.sh"
+# define DEFAULT_URL_VIEWER "viewurl-auto.sh"
 class KURLRequester : public QLineEdit
 {
 public:
