--- licq-1.3.4-RC1/plugins/qt-gui/src/optionsdlg.cpp~	2006-08-19 06:06:49.000000000 +0000
+++ licq-1.3.4-RC1/plugins/qt-gui/src/optionsdlg.cpp	2006-09-21 20:25:37.000000000 +0000
@@ -1503,12 +1503,7 @@
 #ifdef USE_KDE
   cmbUrlViewer->insertItem(DEFAULT_URL_VIEWER);
 #endif
-  cmbUrlViewer->insertItem("viewurl-lynx.sh");
-  cmbUrlViewer->insertItem("viewurl-mozilla.sh");
-  cmbUrlViewer->insertItem("viewurl-ncftp.sh");
-  cmbUrlViewer->insertItem("viewurl-netscape.sh");
-  cmbUrlViewer->insertItem("viewurl-opera.sh");
-  cmbUrlViewer->insertItem("viewurl-w3m.sh");
+  cmbUrlViewer->insertItem("viewurl-auto.sh");
 
   lblTerminal = new QLabel(tr("Terminal:"), boxExtensions);
   edtTerminal = new QLineEdit(tr("Terminal:"), boxExtensions);
