---
 arch/i386/kernel/process.c |    1 -
 1 file changed, 1 deletion(-)

Index: linux-2.6.23.1-rt5/arch/i386/kernel/process.c
===================================================================
--- linux-2.6.23.1-rt5.orig/arch/i386/kernel/process.c
+++ linux-2.6.23.1-rt5/arch/i386/kernel/process.c
@@ -186,7 +186,6 @@ void cpu_idle(void)
 			if (__get_cpu_var(cpu_idle_state))
 				__get_cpu_var(cpu_idle_state) = 0;
 
-			check_pgt_cache();
 			rmb();
 			idle = pm_idle;
 
