--- linux-2.6.34/arch/arm/mach-pxa/spitz.c~	2010-05-26 22:01:01.440705740 -0400
+++ linux-2.6.34/arch/arm/mach-pxa/spitz.c	2010-05-26 22:16:58.552706405 -0400
@@ -629,16 +630,13 @@ static struct pxafb_mach_info spitz_pxaf
 
 static struct mtd_partition sharpsl_nand_partitions[] = {
 	{
-		.name = "System Area",
+		.name = "Bootloader/Emergency Area",
 		.offset = 0,
 		.size = 7 * 1024 * 1024,
 	},
 	{
 		.name = "Root Filesystem",
 		.offset = 7 * 1024 * 1024,
-	},
-	{
-		.name = "Home Filesystem",
 		.offset = MTDPART_OFS_APPEND,
 		.size = MTDPART_SIZ_FULL,
 	},
@@ -738,9 +736,11 @@ static void __init common_init(void)
 	if (machine_is_spitz()) {
 		sharpsl_nand_partitions[1].size = 5 * 1024 * 1024;
 	} else if (machine_is_akita()) {
-		sharpsl_nand_partitions[1].size = 58 * 1024 * 1024;
+		sharpsl_nand_partitions[1].size = 121 * 1024 * 1024;
 	} else if (machine_is_borzoi()) {
-		sharpsl_nand_partitions[1].size = 32 * 1024 * 1024;
+		sharpsl_nand_partitions[1].size = 121 * 1024 * 1024;
+	} else if (machine_is_terrier()) {
+		sharpsl_nand_partitions[1].size = 121 * 1024 * 1024;
 	}
 
 	PMCR = 0x00;
@@ -849,6 +849,7 @@ static void __init fixup_spitz(struct ma
 #ifdef CONFIG_MACH_SPITZ
 MACHINE_START(SPITZ, "SHARP Spitz")
 	.phys_io	= 0x40000000,
+	.boot_params	= 0xa0000100,	/* BLOB boot parameter setting */
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
 	.fixup		= fixup_spitz,
 	.map_io		= pxa_map_io,
@@ -861,6 +862,7 @@ MACHINE_END
 #ifdef CONFIG_MACH_BORZOI
 MACHINE_START(BORZOI, "SHARP Borzoi")
 	.phys_io	= 0x40000000,
+	.boot_params	= 0xa0000100,	/* BLOB boot parameter setting */
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
 	.fixup		= fixup_spitz,
 	.map_io		= pxa_map_io,
@@ -873,6 +875,7 @@ MACHINE_END
 #ifdef CONFIG_MACH_AKITA
 MACHINE_START(AKITA, "SHARP Akita")
 	.phys_io	= 0x40000000,
+	.boot_params	= 0xa0000100,	/* BLOB boot parameter setting */
 	.io_pg_offst	= (io_p2v(0x40000000) >> 18) & 0xfffc,
 	.fixup		= fixup_spitz,
 	.map_io		= pxa_map_io,
