--- mc-4.6.1/edit/editcmd.c~	2006-07-06 12:21:47.000000000 +0000
+++ mc-4.6.1/edit/editcmd.c	2006-07-06 12:26:50.000000000 +0000
@@ -3037,13 +3037,12 @@
 #ifndef UTF8
 	    if (strncmp
 		(&compl[i].text[word_len], &bufpos[word_len],
-		 max (len,
-#else /* UTF8 */
+		 max (len, compl[i].len) - word_len) == 0) {
+#else
             if (wcsncmp
                 ((wchar_t *) &compl[i].text[word_len],
-                 (wchar_t *) &bufpos[word_len], max (len,
-#endif /* UTF8 */
-                 compl[i].len) - word_len) == 0) {
+                 (wchar_t *) &bufpos[word_len], max (len, compl[i].len) - word_len) == 0) {
+#endif
 		skip = 1;
 		break;		/* skip it, already added */
 	    }
