--- s/src/Unix/Linux/sheepthreads.c-	2010-08-18 14:04:05.835348573 +0300
+++ s/src/Unix/Linux/sheepthreads.c	2010-08-18 14:04:59.885314420 +0300
@@ -45,9 +45,6 @@
 extern int atomic_or(int *var, int or);
 extern int test_and_set(int *var, int val);
 
-/* Linux kernel calls */
-extern int __clone(int (*fn)(void *), void *, int, void *);
-
 /* struct sem_t */
 typedef struct {
 	struct _pthread_fastlock __sem_lock;
