--- lynx2-8-5/src/chrtrans/makefile.in~	2006-07-04 13:41:32.000000000 +0000
+++ lynx2-8-5/src/chrtrans/makefile.in	2006-07-04 14:05:00.000000000 +0000
@@ -96,10 +96,8 @@
 
 tables: $(TABLES)
 
-makeuctb$x: makeuctb$o
-	$(CC) $(CC_OPTS) $(LDFLAGS) -o $@ makeuctb$o $(INTLLIB) $(LIBS)
-
-makeuctb$o: $(srcdir)/UCkd.h $(srcdir)/makeuctb.c
+makeuctb$x: 
+	gcc -DHAVE_CONFIG_H -DLINUX -D_GNU_SOURCE -I../.. -I../../src -I../../src/chrtrans -I../../WWW/Library/Implementation -I../../  -o $@ makeuctb.c
 
 .SUFFIXES : $o .tbl .i
 
