--- s/configure-	2009-08-29 03:29:04.000000000 +0700
+++ s/configure	2009-09-06 15:57:34.000000000 +0700
@@ -40535,92 +40535,11 @@
 fi
 
 
-	if test x$mc_slang_is_valid_version = xno; then
-            found_slang=no
-            error_msg_slang="S-Lang library version 2.0 or newer not found"
-	fi
+            found_slang=yes
 	{ $as_echo "$as_me:$LINENO: result: $mc_slang_is_valid_version" >&5
 $as_echo "$mc_slang_is_valid_version" >&6; }
     fi
 
-        if test x"$found_slang" = x"yes"; then
-	unset ac_cv_lib_slang_SLsmg_write_nwchars
-        { $as_echo "$as_me:$LINENO: checking for SLsmg_write_nwchars in -lslang" >&5
-$as_echo_n "checking for SLsmg_write_nwchars in -lslang... " >&6; }
-if test "${ac_cv_lib_slang_SLsmg_write_nwchars+set}" = set; then
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lslang  $LIBS"
-cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char SLsmg_write_nwchars ();
-int
-main ()
-{
-return SLsmg_write_nwchars ();
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext conftest$ac_exeext
-if { (ac_try="$ac_link"
-case "(($ac_try" in
-  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
-  *) ac_try_echo=$ac_try;;
-esac
-eval ac_try_echo="\"\$as_me:$LINENO: $ac_try_echo\""
-$as_echo "$ac_try_echo") >&5
-  (eval "$ac_link") 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && {
-	 test -z "$ac_c_werror_flag" ||
-	 test ! -s conftest.err
-       } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
-	 $as_test_x conftest$ac_exeext
-       }; then
-  ac_cv_lib_slang_SLsmg_write_nwchars=yes
-else
-  $as_echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-	ac_cv_lib_slang_SLsmg_write_nwchars=no
-fi
-
-rm -rf conftest.dSYM
-rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
-      conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:$LINENO: result: $ac_cv_lib_slang_SLsmg_write_nwchars" >&5
-$as_echo "$ac_cv_lib_slang_SLsmg_write_nwchars" >&6; }
-if test "x$ac_cv_lib_slang_SLsmg_write_nwchars" = x""yes; then
-
-                found_slang=no
-                error_msg_slang="Rejecting S-Lang with UTF-8 support, it's not fully supported yet"
-
-else
-  :
-
-fi
-
-    fi
 
     if test x"$found_slang" = x"yes"; then
         screen_type=slang
