--- epdfview-0.1.6/src/IDocumentLink.h.orig	2008-01-23 10:59:05.000000000 +0100
+++ epdfview-0.1.6/src/IDocumentLink.h	2008-01-23 10:59:24.000000000 +0100
@@ -33,7 +33,7 @@
     class IDocumentLink
     {
         public:
-            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y1);
+            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y2);
             virtual ~IDocumentLink (void);
 
             virtual void activate (IDocument *document) = 0;
