--- gst-ffmpeg-0.10.8/gst-libs/ext/ffmpeg/configure-	2009-08-21 01:43:18.000000000 +0700
+++ gst-ffmpeg-0.10.8/gst-libs/ext/ffmpeg/configure	2009-08-21 01:43:29.000000000 +0700
@@ -1604,7 +1604,7 @@
         objformat="coff"
         enable dos_paths
         ;;
-    linux)
+    linux*)
         enable dv1394
         ;;
     irix*)
@@ -1851,6 +1851,8 @@
 enabled ppc && check_asm dcbzl '"dcbzl 0, 1"'
 enabled ppc && check_asm ppc4xx '"maclhw r10, r11, r12"'
 
+enabled ppc && add_cflags -fPIC
+
 # check for SIMD availability
 
 # AltiVec flags: The FSF version of GCC differs from the Apple version
