--- s/devices.c-	2009-01-16 04:56:36.000000000 +0600
+++ s/devices.c	2009-01-16 04:59:54.000000000 +0600
@@ -270,7 +270,7 @@
 	twin_set_file(pboot_proc_server_sock, sock, TWIN_READ, &_ctx);
 
 	if (udev_trigger) {
-		int rc = system("udevtrigger");
+		int rc = system("/sbin/udevadm trigger");
 		if (rc)
 			LOG("udevtrigger failed, rc %d\n", rc);
 	}
@@ -284,7 +284,7 @@
 	char *kexec_opts[10];
 	int i, nr_opts = 2;
 
-	kexec_opts[0] = "/usr/sbin/kexec";
+	kexec_opts[0] = "/sbin/kexec";
 	kexec_opts[1] = "-f";
 	if (opt->initrd_file && *opt->initrd_file) {
 		kexec_opts[nr_opts] = malloc(10 + strlen(opt->initrd_file));
