--- s/Makefile.am~	2006-08-17 16:06:30.000000000 +0000
+++ s/Makefile.am	2006-08-17 16:06:43.000000000 +0000
@@ -1,6 +1,6 @@
 ## Makefile.am for GTK+
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
--- s/Makefile.in~	2006-08-17 16:06:33.000000000 +0000
+++ s/Makefile.in	2006-08-17 16:06:56.000000000 +0000
@@ -292,7 +292,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
