--- linux-2.6.25/drivers/i2c/busses/i2c-davinci.c-	2008-05-07 20:10:28.000000000 +0700
+++ linux-2.6.25/drivers/i2c/busses/i2c-davinci.c	2008-05-08 16:36:31.000000000 +0700
@@ -120,8 +120,8 @@
 
 /* default platform data to use if not supplied in the platform_device */
 static struct davinci_i2c_platform_data davinci_i2c_platform_data_default = {
-	.bus_freq	= 100,
-	.bus_delay	= 0,
+	.bus_freq	= 20,
+	.bus_delay	= 20, /* Introduce a 20musec delay.  Required for Davinci EVM */
 };
 
 static inline void davinci_i2c_write_reg(struct davinci_i2c_dev *i2c_dev,
@@ -175,6 +175,14 @@
 	 *       if PSC > 1 , d = 5
 	 */
 
+	psc = 26; /* To get 1MHz clock */
+
+        clk = ((input_clock/(psc + 1)) / (pdata->bus_freq * 1000)) - 10;
+
+	clkh = (27 * clk) / 100; /* duty cycle should be 27% */
+	clkl = (clk - clkh);
+
+#if 0
 	/* get minimum of 7 MHz clock, but max of 12 MHz */
 	psc = (input_clock / 7000000) - 1;
 	if ((input_clock / (psc + 1)) > 12000000)
@@ -184,6 +192,7 @@
 	clk = ((input_clock / (psc + 1)) / (pdata->bus_freq * 1000)) - (d<<1);
 	clkh = clk>>1;
 	clkl = clk - clkh;
+#endif
 
 	davinci_i2c_write_reg(dev, DAVINCI_I2C_PSC_REG, psc);
 	davinci_i2c_write_reg(dev, DAVINCI_I2C_CLKH_REG, clkh);
