--- linux-2.6.25/arch/arm/mach-davinci/board-evm.c-	2008-05-07 09:56:56.000000000 +0700
+++ linux-2.6.25/arch/arm/mach-davinci/board-evm.c	2008-05-07 09:57:34.000000000 +0700
@@ -101,6 +101,7 @@
 };
 
 #if defined(CONFIG_MTD_NAND_DAVINCI) || defined(CONFIG_MTD_NAND_DAVINCI_MODULE)
+#if 0
 struct mtd_partition davinci_evm_nandflash_partition[] = {
 	/* 5 MB space at the beginning for bootloader and kernel */
 	{
@@ -110,6 +111,58 @@
 		.mask_flags	= 0,
 	}
 };
+#else
+struct mtd_partition davinci_evm_nandflash_partition[] = {                                                                                             
+        { 
+	    .name		= "UBL",                                                                                                                       
+            .offset		= 0,                                                                                                                         
+            .size		= 0x18000,
+	    .mask_flags 	= MTD_WRITEABLE, /* force read-only */
+	},
+        { 
+	    .name		= "U-boot",                                                                                                                    
+            .offset		= MTDPART_OFS_APPEND,                                                                                                        
+            .size		= 0x38000,
+	    .mask_flags 	= MTD_WRITEABLE, /* force read-only */
+	},
+        { 
+	    .name		= "U-boot Env",                                                                                                                
+            .offset		= MTDPART_OFS_APPEND,                                                                                                        
+            .size		= 0x4000,
+	    .mask_flags 	= MTD_WRITEABLE, /* force read-only */
+	},
+        {
+	    .name		= "kernel",                                                                                                                    
+            .offset		= MTDPART_OFS_APPEND,                                                                                                        
+            .size		= 0x400000,
+	    .mask_flags 	= MTD_WRITEABLE, /* force read-only */
+	},
+        { 
+	    .name		= "sysconfig",                                                                                                     
+            .offset		= MTDPART_OFS_APPEND,                                                                                                        
+            .size		= 0xE4000,
+	    .mask_flags 	= MTD_WRITEABLE, /* force read-only */
+	},
+        { 
+	    .name		= "hwconfig",                                                                                                                  
+            .offset		= MTDPART_OFS_APPEND,                                                                                                        
+            .size		= 0x8000,
+	    .mask_flags 	= MTD_WRITEABLE, /* force read-only */
+	},
+        { 
+	    .name		= "local",                                                                                                                     
+            .offset		= MTDPART_OFS_APPEND,                                                                                                        
+            .size		= 0x1800000,
+	    .mask_flags		= 0,
+	},
+        {
+	    .name		= "root",                                                                                                                      
+            .offset		= MTDPART_OFS_APPEND,                                                                                                        
+            .size		= MTDPART_SIZ_FULL,
+	    .mask_flags		= 0,
+	},
+};                                                                                                                                           
+#endif
 
 static struct flash_platform_data davinci_evm_nandflash_data = {
 	.parts		= davinci_evm_nandflash_partition,
