--- s/hw/omap.c-	2009-01-11 13:04:29.000000000 +0600
+++ s/hw/omap.c	2009-01-11 13:18:27.000000000 +0600
@@ -4668,7 +4668,11 @@
         s->round = 0;
     }
 
+#ifdef _WIN32
+    memcpy(&s->current_tm,localtime(&s->ti),sizeof(struct tm));
+#else
     localtime_r(&s->ti, &s->current_tm);
+#endif
 
     if ((s->interrupts & 0x08) && s->ti == s->alarm_ti) {
         s->status |= 0x40;
@@ -4730,8 +4734,12 @@
     s->alarm_tm.tm_mday = 0x01;
     s->status = 1 << 7;
     time(&s->ti);
+#ifdef _WIN32
+    memcpy(&s->current_tm,localtime(&s->ti),sizeof(struct tm));
+    s->ti = mktime(&s->current_tm);
+#else
     s->ti = mktime(s->convert(&s->ti, &s->current_tm));
-
+#endif
     omap_rtc_alarm_update(s);
     omap_rtc_tick(s);
 }
@@ -4747,8 +4755,11 @@
     s->irq = irq[0];
     s->alarm = irq[1];
     s->clk = qemu_new_timer(rt_clock, omap_rtc_tick, s);
+#ifdef _WIN32
+    s->convert = localtime;
+#else
     s->convert = rtc_utc ? gmtime_r : localtime_r;
-
+#endif
     omap_rtc_reset(s);
 
     iomemtype = cpu_register_io_memory(0, omap_rtc_readfn,
