--- s/os-interface.c.orig	2009-03-24 05:37:38.000000000 +0600
+++ s/os-interface.c	2009-12-14 23:10:05.155186313 +0600
@@ -88,7 +88,7 @@
     nv_stack_t        *sp;
     struct completion  completion;
 #if defined(CONFIG_PREEMPT_RT)
-    raw_spinlock_t     lock;
+    atomic_spinlock_t     lock;
 #else
     spinlock_t         lock;
 #endif
@@ -127,7 +127,11 @@
     os_sema = (os_sema_t *)*ppSema;
     os_sema->sp = sp;
     init_completion(&os_sema->completion);
+#if defined(CONFIG_PREEMPT_RT)
+    atomic_spin_lock_init(&os_sema->lock);
+#else
     spin_lock_init(&os_sema->lock);
+#endif
     os_sema->count = 1;
 
     return RM_OK;
@@ -171,18 +175,30 @@
     os_sema_t *os_sema = (os_sema_t *)pSema;
     unsigned long old_irq;
 
+#if defined(CONFIG_PREEMPT_RT)
+    atomic_spin_lock_irqsave(&os_sema->lock, old_irq);
+#else
     spin_lock_irqsave(&os_sema->lock, old_irq);
+#endif
     if (os_sema->count <= 0)
     {
         os_sema->count--;
+#if defined(CONFIG_PREEMPT_RT)
+        atomic_spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#else
         spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#endif
         wait_for_completion(&os_sema->completion);
     }
     else
     {
         os_sema->count--;
         rm_disable_interrupts(os_sema->sp);
+#if defined(CONFIG_PREEMPT_RT)
+        atomic_spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#else
         spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#endif
     }
 
     return RM_OK;
@@ -205,17 +221,29 @@
     os_sema_t *os_sema = (os_sema_t *)pSema;
     unsigned long old_irq;
 
+#if defined(CONFIG_PREEMPT_RT)
+    atomic_spin_lock_irqsave(&os_sema->lock, old_irq);
+#else
     spin_lock_irqsave(&os_sema->lock, old_irq);
+#endif
     if (os_sema->count <= 0)
     {
+#if defined(CONFIG_PREEMPT_RT)
+        atomic_spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#else
         spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#endif
         return FALSE;
     }
     else
     {
         os_sema->count--;
         rm_disable_interrupts(os_sema->sp);
+#if defined(CONFIG_PREEMPT_RT)
+        atomic_spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#else
         spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#endif
         return TRUE;
     }
 
@@ -239,7 +267,11 @@
     unsigned long old_irq;
     BOOL doWakeup;
 
+#if defined(CONFIG_PREEMPT_RT)
+    atomic_spin_lock_irqsave(&os_sema->lock, old_irq);
+#else
     spin_lock_irqsave(&os_sema->lock, old_irq);
+#endif
     if (os_sema->count < 0)
     {
         doWakeup = TRUE;
@@ -250,7 +282,11 @@
         rm_enable_interrupts(os_sema->sp);
     }
     os_sema->count++;
+#if defined(CONFIG_PREEMPT_RT)
+    atomic_spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#else
     spin_unlock_irqrestore(&os_sema->lock, old_irq);
+#endif
 
     if (doWakeup)
         complete(&os_sema->completion);
