--- s/configure.in.orig	2009-12-02 22:01:42.000000000 +0600
+++ s/configure.in	2009-12-04 16:25:26.515237384 +0600
@@ -41,48 +41,9 @@
 
 
 dnl libparted
-LIBPARTED_VERSION=1.7.1
-AC_MSG_CHECKING(for libparted >= $LIBPARTED_VERSION)
-LIBS_save="$LIBS"
-LIBS="-lparted -luuid -ldl"
-AC_TRY_RUN(
-#include <stdio.h>
-#include <parted/parted.h>
-
-int main ()
-{
-	int min_major ;
-	int min_minor ;
-	int min_micro ;
-	int major ;
-	int minor ;
-	int micro ;
-
-	if ( sscanf( "$LIBPARTED_VERSION", "%d.%d.%d", &min_major, &min_minor, &min_micro ) == 3 )
-	{
-		/* Check for libparted three digit version number */
-		if ( sscanf( ped_get_version(), "%d.%d.%d", &major, &minor, &micro ) == 3 )
-		{
-			printf( "Found libparted %s\t", ped_get_version() ) ;
-			return ! ( (major > min_major) ||
-			           ((major == min_major) && (minor > min_minor)) ||
-			           ((major == min_major) && (minor == min_minor) && (micro >= min_micro)) ) ;
-		}
-
-		/* Check for two digit libparted version number */
-		if ( sscanf( ped_get_version(), "%d.%d", &major, &minor ) == 2 )
-		{
-			printf( "Found libparted %s\t", ped_get_version() ) ;
-			return ! ( (major > min_major) ||
-			           ((major == min_major) && (minor >= min_minor)) ) ;
-		}
-	}
-
-	return 1 ;
-}
-,AC_MSG_RESULT(OK),AC_MSG_ERROR(*** Requires libparted >= $LIBPARTED_VERSION.  Perhaps development header files missing?) )
-LIBS="$LIBS_save"
-
+PKG_CHECK_MODULES(LIBPARTED, libparted > 1.7.1 );
+LIBS="$LIBS $LIBPARTED_LIBS"
+CFLAGS="$CFLAGS $LIBPARTED_CFLAGS"
 
 dnl GTKMM
 PKG_CHECK_MODULES(GTKMM, gtkmm-2.4 > 2.8 )
