--- libnsfb-svn/src/surface/Makefile-	2010-06-04 18:46:26.277281583 +0700
+++ libnsfb-svn/src/surface/Makefile	2010-06-04 18:46:29.867866014 +0700
@@ -1,4 +1,4 @@
 # Sources
-DIR_SOURCES := surface.c sdl.c vnc.c able.c ram.c linux.c x.c
+DIR_SOURCES := surface.c sdl.c vnc.c able.c ram.c linux.c
 
 include build/makefiles/Makefile.subdir
--- libnsfb-svn/libnsfb.pc.in-	2010-06-04 19:00:20.417846112 +0700
+++ libnsfb-svn/libnsfb.pc.in	2010-06-04 19:00:36.116596082 +0700
@@ -6,6 +6,6 @@
 Name: libnsfb
 Description: Provides framebuffer access for netsurf.
 Version: VERSION
-Requires: sdl xcb xcb-icccm xcb-image xcb-keysyms
+Requires: sdl
 Libs: -L${libdir} -lnsfb
 Cflags: -I${includedir}
