Index: devices.c
===================================================================
--- s/devices.c	(revision 872)
+++ s/devices.c	(working copy)
@@ -121,7 +121,9 @@
     int rc = system(buf);
     if (rc)
 	fprintf(stderr, "mkdir problem\n");
-    
+
+    bootdevice_remove(dev_path);
+
     sprintf(buf, MOUNT_BIN " -o ro %s " MOUNT_DIR "%s", dev_path, dev_path);
     rc = system(buf);
     if (!rc) {
