--- s/pm/functions.in-	2009-01-12 20:22:32.000000000 +0600
+++ s/pm/functions.in	2009-01-12 20:40:53.000000000 +0600
@@ -208,12 +208,13 @@
 	if [ "$1" = "all" ]; then
 	    echo '' > "$PARAMETERS.new"
 	else
-	    echo '' >"$PARAMETERS.rm"
+	    echo -n '' >"$PARAMETERS.rm"
 	    for p in "$@"; do
-		echo "$p" >> "$PARAMETERS.rm"
+		echo -n "^${p}\$|" >> "$PARAMETERS.rm"
 	    done
+	    echo -n "^${p}\$" >> "$PARAMETERS.rm"
 	    # let grep do the dirty work.
-	    grep -vxFf "$PARAMETERS.rm" "$PARAMETERS" > "$PARAMETERS.new"
+	    grep -vEf "$PARAMETERS.rm" "$PARAMETERS" > "$PARAMETERS.new"
 	fi
 	cp -f "$PARAMETERS.new" "$PARAMETERS"
 }
