diff -Nur MPlayer-1.0pre8.orig/configure MPlayer-1.0pre8/configure
--- MPlayer-1.0pre8.orig/configure	2006-06-11 18:35:47.000000000 +0000
+++ MPlayer-1.0pre8/configure	2006-07-03 12:45:56.000000000 +0000
@@ -135,7 +135,7 @@
 
 ppc() {
   case "$host_arch" in
-    ppc) return 0;;
+    ppc|powerpc) return 0;;
     *) return 1;;
   esac
 }
@@ -559,7 +559,7 @@
   # host's CPU/instruction set
    host_arch=`uname -p 2>&1`
    case "$host_arch" in
-   i386|sparc|ppc|alpha|arm|mips|vax)
+   i386|sparc|ppc|powerpc|alpha|arm|mips|vax)
      ;;
    powerpc) # Darwin returns 'powerpc'
      host_arch=ppc
@@ -582,7 +582,7 @@
           host_arch=i386
         fi
       ;;
-      macppc|ppc|ppc64) host_arch=ppc ;;
+      macppc|ppc|powerpc|ppc64) host_arch=ppc ;;
       alpha) host_arch=alpha ;;
       sparc) host_arch=sparc ;;
       sparc64) host_arch=sparc64 ;;
@@ -1182,7 +1182,7 @@
     _optimizing=''
     ;;
 
-  ppc)
+  ppc|powerpc)
     _def_arch='#define ARCH_POWERPC 1'
     _def_dcbzl='#define NO_DCBZL 1'
     _target_arch='TARGET_ARCH_POWERPC = yes'
