--- fontconfig-2.6.0/fc-case/Makefile.am-	2008-12-11 01:11:11.000000000 +0600
+++ fontconfig-2.6.0/fc-case/Makefile.am	2008-12-11 01:11:44.000000000 +0600
@@ -24,7 +24,7 @@
 
 CC = @CC_FOR_BUILD@
 EXEEXT = @EXEEXT_FOR_BUILD@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) -o $@
 
 INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(WARN_CFLAGS)
 
--- fontconfig-2.6.0/fc-case/Makefile.in-	2008-12-11 01:11:15.000000000 +0600
+++ fontconfig-2.6.0/fc-case/Makefile.in	2008-12-11 01:12:08.000000000 +0600
@@ -242,7 +242,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) -o $@
 INCLUDES = -I${top_srcdir}/src -I${top_srcdir} $(WARN_CFLAGS)
 TMPL = fccase.tmpl.h
 STMPL = ${top_srcdir}/fc-case/${TMPL}
