--- mplayerplug-in-3.55/Source/plugin-list.cpp-	2010-01-18 16:59:11.376591511 +0600
+++ mplayerplug-in-3.55/Source/plugin-list.cpp	2010-01-18 17:03:15.365963565 +0600
@@ -140,8 +140,8 @@
     char tagtarget[128];
     char tagurl[4096];
 
-    endvideo = strstr(smilbuffer, "</video");
-    startarea = strstr(smilbuffer, "<area");
+    endvideo = strstr((char *) smilbuffer, "</video");
+    startarea = strstr((char *) smilbuffer, "<area");
 
     while (startarea && startarea < endvideo) {
 	start = strstr(startarea + 5, "begin=");
