--- s/configure-	2009-01-04 14:59:02.000000000 +0600
+++ s/configure	2009-01-04 15:00:27.000000000 +0600
@@ -21707,10 +21707,6 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-$as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "${sdkdir}/dri.h"; then
   eval "$as_ac_File=yes"
 else
@@ -21735,10 +21731,6 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-$as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "${sdkdir}/sarea.h"; then
   eval "$as_ac_File=yes"
 else
@@ -21763,10 +21755,6 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-$as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "${sdkdir}/dristruct.h"; then
   eval "$as_ac_File=yes"
 else
