--- mozilla/js/src/Makefile.in.orig	2008-05-29 01:42:32.000000000 +0700
+++ mozilla/js/src/Makefile.in	2008-12-29 20:10:28.000000000 +0600
@@ -393,10 +393,8 @@
 jsautocfg.h:
 	touch $@
 else
-jsautocfg.h: jscpucfg$(HOST_BIN_SUFFIX)
-	@rm -f $@ jsautocfg.tmp
-	./jscpucfg > jsautocfg.tmp
-	mv jsautocfg.tmp $@
+jsautocfg.h:
+	touch $@
 endif
 endif
 
@@ -417,14 +415,12 @@
 # #ifdef i386.  This macro is among the NSPR defines, but is also automatically
 # defined by the compiler when building for i386.  It therefore needs to be
 # defined here if targeting i386, and explicitly undefined otherwise.
-ifeq ($(OS_ARCH),Darwin)
 ifeq ($(TARGET_CPU),powerpc)
 JSCPUCFG_DEFINES += -Ui386
 else
 JSCPUCFG_DEFINES += -Di386=1
 endif
 endif
-endif
 
 ifeq ($(OS_ARCH),QNX)
 ifneq ($(OS_TARGET),NTO)
