--- s/vl.c~	2006-08-25 11:18:41.000000000 +0000
+++ s/vl.c	2006-08-25 11:18:57.000000000 +0000
@@ -541,7 +541,7 @@
 static void init_get_clock(void)
 {
     use_rt_clock = 0;
-#if defined(__linux__)
+#if defined(__linux__) && defined(_POSIX_MONOTONIC_CLOCK)
     {
         struct timespec ts;
         if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) {
@@ -553,7 +553,7 @@
 
 static int64_t get_clock(void)
 {
-#if defined(__linux__)
+#if defined(__linux__) && defined(_POSIX_MONOTONIC_CLOCK)
     if (use_rt_clock) {
         struct timespec ts;
         clock_gettime(CLOCK_MONOTONIC, &ts);
