--- geany-0.18/src/keyfile.c-	2009-07-20 19:25:24.000000000 +0700
+++ geany-0.18/src/keyfile.c	2009-11-25 14:08:15.000000000 +0600
@@ -72,9 +72,9 @@
 #ifdef G_OS_WIN32
 #define GEANY_DEFAULT_TOOLS_TERMINAL	"cmd.exe"
 #else
-#define GEANY_DEFAULT_TOOLS_TERMINAL	"xterm"
+#define GEANY_DEFAULT_TOOLS_TERMINAL	"xterminal"
 #endif
-#define GEANY_DEFAULT_TOOLS_BROWSER		"firefox"
+#define GEANY_DEFAULT_TOOLS_BROWSER		"webbrowser"
 #define GEANY_DEFAULT_TOOLS_PRINTCMD	"lpr"
 #define GEANY_DEFAULT_TOOLS_GREP		"grep"
 #define GEANY_DEFAULT_MRU_LENGTH		10
@@ -725,7 +725,7 @@
 
 		vc->emulation = utils_get_setting_string(config, "VTE", "emulation", "xterm");
 		vc->shell = utils_get_setting_string(config, "VTE", "shell", shell);
-		vc->font = utils_get_setting_string(config, "VTE", "font", "Monospace 10");
+		vc->font = utils_get_setting_string(config, "VTE", "font", "Fixed 10");
 		vc->scroll_on_key = utils_get_setting_boolean(config, "VTE", "scroll_on_key", TRUE);
 		vc->scroll_on_out = utils_get_setting_boolean(config, "VTE", "scroll_on_out", TRUE);
 		vc->enable_bash_keys = utils_get_setting_boolean(config, "VTE", "enable_bash_keys", FALSE);
